// @aws-sdk/client-iam/dist-es/commands/CreateServiceSpecificCredentialCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateServiceSpecificCredentialRequest, CreateServiceSpecificCredentialResponse } from "../models/models_0";
import { deserializeAws_queryCreateServiceSpecificCredentialCommand, serializeAws_queryCreateServiceSpecificCredentialCommand, } from "../protocols/Aws_query";
var CreateServiceSpecificCredentialCommand = (function (_super) {
    __extends(CreateServiceSpecificCredentialCommand, _super);
    function CreateServiceSpecificCredentialCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateServiceSpecificCredentialCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "CreateServiceSpecificCredentialCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateServiceSpecificCredentialRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateServiceSpecificCredentialResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateServiceSpecificCredentialCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateServiceSpecificCredentialCommand(input, context);
    };
    CreateServiceSpecificCredentialCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateServiceSpecificCredentialCommand(output, context);
    };
    return CreateServiceSpecificCredentialCommand;
}($Command));
export { CreateServiceSpecificCredentialCommand };
