// @aws-sdk/client-iam/dist-es/commands/CreatePolicyVersionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreatePolicyVersionRequest, CreatePolicyVersionResponse } from "../models/models_0";
import { deserializeAws_queryCreatePolicyVersionCommand, serializeAws_queryCreatePolicyVersionCommand, } from "../protocols/Aws_query";
var CreatePolicyVersionCommand = (function (_super) {
    __extends(CreatePolicyVersionCommand, _super);
    function CreatePolicyVersionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreatePolicyVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "CreatePolicyVersionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreatePolicyVersionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreatePolicyVersionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreatePolicyVersionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreatePolicyVersionCommand(input, context);
    };
    CreatePolicyVersionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreatePolicyVersionCommand(output, context);
    };
    return CreatePolicyVersionCommand;
}($Command));
export { CreatePolicyVersionCommand };
