// @aws-sdk/client-iam/dist-es/commands/CreateServiceLinkedRoleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateServiceLinkedRoleRequest, CreateServiceLinkedRoleResponse } from "../models/models_0";
import { deserializeAws_queryCreateServiceLinkedRoleCommand, serializeAws_queryCreateServiceLinkedRoleCommand, } from "../protocols/Aws_query";
var CreateServiceLinkedRoleCommand = (function (_super) {
    __extends(CreateServiceLinkedRoleCommand, _super);
    function CreateServiceLinkedRoleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateServiceLinkedRoleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "CreateServiceLinkedRoleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateServiceLinkedRoleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateServiceLinkedRoleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateServiceLinkedRoleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateServiceLinkedRoleCommand(input, context);
    };
    CreateServiceLinkedRoleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateServiceLinkedRoleCommand(output, context);
    };
    return CreateServiceLinkedRoleCommand;
}($Command));
export { CreateServiceLinkedRoleCommand };
