// @aws-sdk/client-iam/dist-es/commands/CreateRoleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateRoleRequest, CreateRoleResponse } from "../models/models_0";
import { deserializeAws_queryCreateRoleCommand, serializeAws_queryCreateRoleCommand } from "../protocols/Aws_query";
var CreateRoleCommand = (function (_super) {
    __extends(CreateRoleCommand, _super);
    function CreateRoleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateRoleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "CreateRoleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateRoleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateRoleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateRoleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateRoleCommand(input, context);
    };
    CreateRoleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateRoleCommand(output, context);
    };
    return CreateRoleCommand;
}($Command));
export { CreateRoleCommand };
