// @aws-sdk/client-iam/dist-es/commands/CreateOpenIDConnectProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateOpenIDConnectProviderRequest, CreateOpenIDConnectProviderResponse } from "../models/models_0";
import { deserializeAws_queryCreateOpenIDConnectProviderCommand, serializeAws_queryCreateOpenIDConnectProviderCommand, } from "../protocols/Aws_query";
var CreateOpenIDConnectProviderCommand = (function (_super) {
    __extends(CreateOpenIDConnectProviderCommand, _super);
    function CreateOpenIDConnectProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateOpenIDConnectProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "CreateOpenIDConnectProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateOpenIDConnectProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateOpenIDConnectProviderResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateOpenIDConnectProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateOpenIDConnectProviderCommand(input, context);
    };
    CreateOpenIDConnectProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateOpenIDConnectProviderCommand(output, context);
    };
    return CreateOpenIDConnectProviderCommand;
}($Command));
export { CreateOpenIDConnectProviderCommand };
