// @aws-sdk/client-iam/dist-es/commands/DeactivateMFADeviceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeactivateMFADeviceRequest } from "../models/models_0";
import { deserializeAws_queryDeactivateMFADeviceCommand, serializeAws_queryDeactivateMFADeviceCommand, } from "../protocols/Aws_query";
var DeactivateMFADeviceCommand = (function (_super) {
    __extends(DeactivateMFADeviceCommand, _super);
    function DeactivateMFADeviceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeactivateMFADeviceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeactivateMFADeviceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeactivateMFADeviceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeactivateMFADeviceCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeactivateMFADeviceCommand(input, context);
    };
    DeactivateMFADeviceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeactivateMFADeviceCommand(output, context);
    };
    return DeactivateMFADeviceCommand;
}($Command));
export { DeactivateMFADeviceCommand };
