// @aws-sdk/client-iam/dist-es/commands/DeleteServiceLinkedRoleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteServiceLinkedRoleRequest, DeleteServiceLinkedRoleResponse } from "../models/models_0";
import { deserializeAws_queryDeleteServiceLinkedRoleCommand, serializeAws_queryDeleteServiceLinkedRoleCommand, } from "../protocols/Aws_query";
var DeleteServiceLinkedRoleCommand = (function (_super) {
    __extends(DeleteServiceLinkedRoleCommand, _super);
    function DeleteServiceLinkedRoleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteServiceLinkedRoleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteServiceLinkedRoleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteServiceLinkedRoleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteServiceLinkedRoleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteServiceLinkedRoleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteServiceLinkedRoleCommand(input, context);
    };
    DeleteServiceLinkedRoleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteServiceLinkedRoleCommand(output, context);
    };
    return DeleteServiceLinkedRoleCommand;
}($Command));
export { DeleteServiceLinkedRoleCommand };
