// @aws-sdk/client-iam/dist-es/commands/DeleteOpenIDConnectProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteOpenIDConnectProviderRequest } from "../models/models_0";
import { deserializeAws_queryDeleteOpenIDConnectProviderCommand, serializeAws_queryDeleteOpenIDConnectProviderCommand, } from "../protocols/Aws_query";
var DeleteOpenIDConnectProviderCommand = (function (_super) {
    __extends(DeleteOpenIDConnectProviderCommand, _super);
    function DeleteOpenIDConnectProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteOpenIDConnectProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteOpenIDConnectProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteOpenIDConnectProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteOpenIDConnectProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteOpenIDConnectProviderCommand(input, context);
    };
    DeleteOpenIDConnectProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteOpenIDConnectProviderCommand(output, context);
    };
    return DeleteOpenIDConnectProviderCommand;
}($Command));
export { DeleteOpenIDConnectProviderCommand };
