// @aws-sdk/client-iam/dist-es/commands/DeleteRolePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteRolePolicyRequest } from "../models/models_0";
import { deserializeAws_queryDeleteRolePolicyCommand, serializeAws_queryDeleteRolePolicyCommand, } from "../protocols/Aws_query";
var DeleteRolePolicyCommand = (function (_super) {
    __extends(DeleteRolePolicyCommand, _super);
    function DeleteRolePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteRolePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteRolePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteRolePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteRolePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteRolePolicyCommand(input, context);
    };
    DeleteRolePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteRolePolicyCommand(output, context);
    };
    return DeleteRolePolicyCommand;
}($Command));
export { DeleteRolePolicyCommand };
