// @aws-sdk/client-iam/dist-es/commands/DeleteInstanceProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteInstanceProfileRequest } from "../models/models_0";
import { deserializeAws_queryDeleteInstanceProfileCommand, serializeAws_queryDeleteInstanceProfileCommand, } from "../protocols/Aws_query";
var DeleteInstanceProfileCommand = (function (_super) {
    __extends(DeleteInstanceProfileCommand, _super);
    function DeleteInstanceProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteInstanceProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteInstanceProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteInstanceProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteInstanceProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteInstanceProfileCommand(input, context);
    };
    DeleteInstanceProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteInstanceProfileCommand(output, context);
    };
    return DeleteInstanceProfileCommand;
}($Command));
export { DeleteInstanceProfileCommand };
