// @aws-sdk/client-iam/dist-es/commands/DeleteAccountPasswordPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { deserializeAws_queryDeleteAccountPasswordPolicyCommand, serializeAws_queryDeleteAccountPasswordPolicyCommand, } from "../protocols/Aws_query";
var DeleteAccountPasswordPolicyCommand = (function (_super) {
    __extends(DeleteAccountPasswordPolicyCommand, _super);
    function DeleteAccountPasswordPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteAccountPasswordPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteAccountPasswordPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: function (input) { return input; },
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteAccountPasswordPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteAccountPasswordPolicyCommand(input, context);
    };
    DeleteAccountPasswordPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteAccountPasswordPolicyCommand(output, context);
    };
    return DeleteAccountPasswordPolicyCommand;
}($Command));
export { DeleteAccountPasswordPolicyCommand };
