// @aws-sdk/client-iam/dist-es/commands/DeleteAccessKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteAccessKeyRequest } from "../models/models_0";
import { deserializeAws_queryDeleteAccessKeyCommand, serializeAws_queryDeleteAccessKeyCommand, } from "../protocols/Aws_query";
var DeleteAccessKeyCommand = (function (_super) {
    __extends(DeleteAccessKeyCommand, _super);
    function DeleteAccessKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteAccessKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteAccessKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteAccessKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteAccessKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteAccessKeyCommand(input, context);
    };
    DeleteAccessKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteAccessKeyCommand(output, context);
    };
    return DeleteAccessKeyCommand;
}($Command));
export { DeleteAccessKeyCommand };
