// @aws-sdk/client-iam/dist-es/commands/DeleteSAMLProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteSAMLProviderRequest } from "../models/models_0";
import { deserializeAws_queryDeleteSAMLProviderCommand, serializeAws_queryDeleteSAMLProviderCommand, } from "../protocols/Aws_query";
var DeleteSAMLProviderCommand = (function (_super) {
    __extends(DeleteSAMLProviderCommand, _super);
    function DeleteSAMLProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteSAMLProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteSAMLProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteSAMLProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteSAMLProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteSAMLProviderCommand(input, context);
    };
    DeleteSAMLProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteSAMLProviderCommand(output, context);
    };
    return DeleteSAMLProviderCommand;
}($Command));
export { DeleteSAMLProviderCommand };
