// @aws-sdk/client-iam/dist-es/commands/DeleteLoginProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteLoginProfileRequest } from "../models/models_0";
import { deserializeAws_queryDeleteLoginProfileCommand, serializeAws_queryDeleteLoginProfileCommand, } from "../protocols/Aws_query";
var DeleteLoginProfileCommand = (function (_super) {
    __extends(DeleteLoginProfileCommand, _super);
    function DeleteLoginProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteLoginProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DeleteLoginProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteLoginProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteLoginProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteLoginProfileCommand(input, context);
    };
    DeleteLoginProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteLoginProfileCommand(output, context);
    };
    return DeleteLoginProfileCommand;
}($Command));
export { DeleteLoginProfileCommand };
