// @aws-sdk/client-iam/dist-es/commands/DetachRolePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetachRolePolicyRequest } from "../models/models_0";
import { deserializeAws_queryDetachRolePolicyCommand, serializeAws_queryDetachRolePolicyCommand, } from "../protocols/Aws_query";
var DetachRolePolicyCommand = (function (_super) {
    __extends(DetachRolePolicyCommand, _super);
    function DetachRolePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DetachRolePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "DetachRolePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetachRolePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetachRolePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDetachRolePolicyCommand(input, context);
    };
    DetachRolePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDetachRolePolicyCommand(output, context);
    };
    return DetachRolePolicyCommand;
}($Command));
export { DetachRolePolicyCommand };
