// @aws-sdk/client-iam/dist-es/commands/GenerateServiceLastAccessedDetailsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GenerateServiceLastAccessedDetailsRequest, GenerateServiceLastAccessedDetailsResponse, } from "../models/models_0";
import { deserializeAws_queryGenerateServiceLastAccessedDetailsCommand, serializeAws_queryGenerateServiceLastAccessedDetailsCommand, } from "../protocols/Aws_query";
var GenerateServiceLastAccessedDetailsCommand = (function (_super) {
    __extends(GenerateServiceLastAccessedDetailsCommand, _super);
    function GenerateServiceLastAccessedDetailsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GenerateServiceLastAccessedDetailsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GenerateServiceLastAccessedDetailsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GenerateServiceLastAccessedDetailsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GenerateServiceLastAccessedDetailsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GenerateServiceLastAccessedDetailsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGenerateServiceLastAccessedDetailsCommand(input, context);
    };
    GenerateServiceLastAccessedDetailsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGenerateServiceLastAccessedDetailsCommand(output, context);
    };
    return GenerateServiceLastAccessedDetailsCommand;
}($Command));
export { GenerateServiceLastAccessedDetailsCommand };
