// @aws-sdk/client-iam/dist-es/commands/GetAccountAuthorizationDetailsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetAccountAuthorizationDetailsRequest, GetAccountAuthorizationDetailsResponse } from "../models/models_0";
import { deserializeAws_queryGetAccountAuthorizationDetailsCommand, serializeAws_queryGetAccountAuthorizationDetailsCommand, } from "../protocols/Aws_query";
var GetAccountAuthorizationDetailsCommand = (function (_super) {
    __extends(GetAccountAuthorizationDetailsCommand, _super);
    function GetAccountAuthorizationDetailsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetAccountAuthorizationDetailsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetAccountAuthorizationDetailsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetAccountAuthorizationDetailsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetAccountAuthorizationDetailsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetAccountAuthorizationDetailsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetAccountAuthorizationDetailsCommand(input, context);
    };
    GetAccountAuthorizationDetailsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetAccountAuthorizationDetailsCommand(output, context);
    };
    return GetAccountAuthorizationDetailsCommand;
}($Command));
export { GetAccountAuthorizationDetailsCommand };
