// @aws-sdk/client-iam/dist-es/commands/GetAccessKeyLastUsedCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetAccessKeyLastUsedRequest, GetAccessKeyLastUsedResponse } from "../models/models_0";
import { deserializeAws_queryGetAccessKeyLastUsedCommand, serializeAws_queryGetAccessKeyLastUsedCommand, } from "../protocols/Aws_query";
var GetAccessKeyLastUsedCommand = (function (_super) {
    __extends(GetAccessKeyLastUsedCommand, _super);
    function GetAccessKeyLastUsedCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetAccessKeyLastUsedCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetAccessKeyLastUsedCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetAccessKeyLastUsedRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetAccessKeyLastUsedResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetAccessKeyLastUsedCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetAccessKeyLastUsedCommand(input, context);
    };
    GetAccessKeyLastUsedCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetAccessKeyLastUsedCommand(output, context);
    };
    return GetAccessKeyLastUsedCommand;
}($Command));
export { GetAccessKeyLastUsedCommand };
