// @aws-sdk/client-iam/dist-es/commands/GetAccountPasswordPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetAccountPasswordPolicyResponse } from "../models/models_0";
import { deserializeAws_queryGetAccountPasswordPolicyCommand, serializeAws_queryGetAccountPasswordPolicyCommand, } from "../protocols/Aws_query";
var GetAccountPasswordPolicyCommand = (function (_super) {
    __extends(GetAccountPasswordPolicyCommand, _super);
    function GetAccountPasswordPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetAccountPasswordPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetAccountPasswordPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: function (input) { return input; },
            outputFilterSensitiveLog: GetAccountPasswordPolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetAccountPasswordPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetAccountPasswordPolicyCommand(input, context);
    };
    GetAccountPasswordPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetAccountPasswordPolicyCommand(output, context);
    };
    return GetAccountPasswordPolicyCommand;
}($Command));
export { GetAccountPasswordPolicyCommand };
