// @aws-sdk/client-iam/dist-es/commands/GetContextKeysForCustomPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetContextKeysForCustomPolicyRequest, GetContextKeysForPolicyResponse } from "../models/models_0";
import { deserializeAws_queryGetContextKeysForCustomPolicyCommand, serializeAws_queryGetContextKeysForCustomPolicyCommand, } from "../protocols/Aws_query";
var GetContextKeysForCustomPolicyCommand = (function (_super) {
    __extends(GetContextKeysForCustomPolicyCommand, _super);
    function GetContextKeysForCustomPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetContextKeysForCustomPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetContextKeysForCustomPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetContextKeysForCustomPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetContextKeysForPolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetContextKeysForCustomPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetContextKeysForCustomPolicyCommand(input, context);
    };
    GetContextKeysForCustomPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetContextKeysForCustomPolicyCommand(output, context);
    };
    return GetContextKeysForCustomPolicyCommand;
}($Command));
export { GetContextKeysForCustomPolicyCommand };
