// @aws-sdk/client-iam/dist-es/commands/GetInstanceProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetInstanceProfileRequest, GetInstanceProfileResponse } from "../models/models_0";
import { deserializeAws_queryGetInstanceProfileCommand, serializeAws_queryGetInstanceProfileCommand, } from "../protocols/Aws_query";
var GetInstanceProfileCommand = (function (_super) {
    __extends(GetInstanceProfileCommand, _super);
    function GetInstanceProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetInstanceProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetInstanceProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetInstanceProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetInstanceProfileResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetInstanceProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetInstanceProfileCommand(input, context);
    };
    GetInstanceProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetInstanceProfileCommand(output, context);
    };
    return GetInstanceProfileCommand;
}($Command));
export { GetInstanceProfileCommand };
