// @aws-sdk/client-iam/dist-es/commands/GetServiceLastAccessedDetailsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetServiceLastAccessedDetailsRequest, GetServiceLastAccessedDetailsResponse } from "../models/models_0";
import { deserializeAws_queryGetServiceLastAccessedDetailsCommand, serializeAws_queryGetServiceLastAccessedDetailsCommand, } from "../protocols/Aws_query";
var GetServiceLastAccessedDetailsCommand = (function (_super) {
    __extends(GetServiceLastAccessedDetailsCommand, _super);
    function GetServiceLastAccessedDetailsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetServiceLastAccessedDetailsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetServiceLastAccessedDetailsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetServiceLastAccessedDetailsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetServiceLastAccessedDetailsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetServiceLastAccessedDetailsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetServiceLastAccessedDetailsCommand(input, context);
    };
    GetServiceLastAccessedDetailsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetServiceLastAccessedDetailsCommand(output, context);
    };
    return GetServiceLastAccessedDetailsCommand;
}($Command));
export { GetServiceLastAccessedDetailsCommand };
