// @aws-sdk/client-iam/dist-es/commands/GetSSHPublicKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetSSHPublicKeyRequest, GetSSHPublicKeyResponse } from "../models/models_0";
import { deserializeAws_queryGetSSHPublicKeyCommand, serializeAws_queryGetSSHPublicKeyCommand, } from "../protocols/Aws_query";
var GetSSHPublicKeyCommand = (function (_super) {
    __extends(GetSSHPublicKeyCommand, _super);
    function GetSSHPublicKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetSSHPublicKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetSSHPublicKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetSSHPublicKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetSSHPublicKeyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetSSHPublicKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetSSHPublicKeyCommand(input, context);
    };
    GetSSHPublicKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetSSHPublicKeyCommand(output, context);
    };
    return GetSSHPublicKeyCommand;
}($Command));
export { GetSSHPublicKeyCommand };
