// @aws-sdk/client-iam/dist-es/commands/GetUserPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetUserPolicyRequest, GetUserPolicyResponse } from "../models/models_0";
import { deserializeAws_queryGetUserPolicyCommand, serializeAws_queryGetUserPolicyCommand, } from "../protocols/Aws_query";
var GetUserPolicyCommand = (function (_super) {
    __extends(GetUserPolicyCommand, _super);
    function GetUserPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetUserPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "GetUserPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetUserPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetUserPolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetUserPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetUserPolicyCommand(input, context);
    };
    GetUserPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetUserPolicyCommand(output, context);
    };
    return GetUserPolicyCommand;
}($Command));
export { GetUserPolicyCommand };
