// @aws-sdk/client-iam/dist-es/commands/ListRolePoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListRolePoliciesRequest, ListRolePoliciesResponse } from "../models/models_0";
import { deserializeAws_queryListRolePoliciesCommand, serializeAws_queryListRolePoliciesCommand, } from "../protocols/Aws_query";
var ListRolePoliciesCommand = (function (_super) {
    __extends(ListRolePoliciesCommand, _super);
    function ListRolePoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListRolePoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListRolePoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListRolePoliciesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListRolePoliciesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListRolePoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListRolePoliciesCommand(input, context);
    };
    ListRolePoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListRolePoliciesCommand(output, context);
    };
    return ListRolePoliciesCommand;
}($Command));
export { ListRolePoliciesCommand };
