// @aws-sdk/client-iam/dist-es/commands/ListSAMLProvidersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSAMLProvidersRequest, ListSAMLProvidersResponse } from "../models/models_0";
import { deserializeAws_queryListSAMLProvidersCommand, serializeAws_queryListSAMLProvidersCommand, } from "../protocols/Aws_query";
var ListSAMLProvidersCommand = (function (_super) {
    __extends(ListSAMLProvidersCommand, _super);
    function ListSAMLProvidersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSAMLProvidersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListSAMLProvidersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSAMLProvidersRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListSAMLProvidersResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSAMLProvidersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListSAMLProvidersCommand(input, context);
    };
    ListSAMLProvidersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListSAMLProvidersCommand(output, context);
    };
    return ListSAMLProvidersCommand;
}($Command));
export { ListSAMLProvidersCommand };
