// @aws-sdk/client-iam/dist-es/commands/ListOpenIDConnectProvidersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListOpenIDConnectProvidersRequest, ListOpenIDConnectProvidersResponse } from "../models/models_0";
import { deserializeAws_queryListOpenIDConnectProvidersCommand, serializeAws_queryListOpenIDConnectProvidersCommand, } from "../protocols/Aws_query";
var ListOpenIDConnectProvidersCommand = (function (_super) {
    __extends(ListOpenIDConnectProvidersCommand, _super);
    function ListOpenIDConnectProvidersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListOpenIDConnectProvidersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListOpenIDConnectProvidersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListOpenIDConnectProvidersRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListOpenIDConnectProvidersResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListOpenIDConnectProvidersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListOpenIDConnectProvidersCommand(input, context);
    };
    ListOpenIDConnectProvidersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListOpenIDConnectProvidersCommand(output, context);
    };
    return ListOpenIDConnectProvidersCommand;
}($Command));
export { ListOpenIDConnectProvidersCommand };
