// @aws-sdk/client-iam/dist-es/commands/ListRolesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListRolesRequest, ListRolesResponse } from "../models/models_0";
import { deserializeAws_queryListRolesCommand, serializeAws_queryListRolesCommand } from "../protocols/Aws_query";
var ListRolesCommand = (function (_super) {
    __extends(ListRolesCommand, _super);
    function ListRolesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListRolesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListRolesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListRolesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListRolesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListRolesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListRolesCommand(input, context);
    };
    ListRolesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListRolesCommand(output, context);
    };
    return ListRolesCommand;
}($Command));
export { ListRolesCommand };
