// @aws-sdk/client-iam/dist-es/commands/ListPoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPoliciesRequest, ListPoliciesResponse } from "../models/models_0";
import { deserializeAws_queryListPoliciesCommand, serializeAws_queryListPoliciesCommand } from "../protocols/Aws_query";
var ListPoliciesCommand = (function (_super) {
    __extends(ListPoliciesCommand, _super);
    function ListPoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListPoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPoliciesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPoliciesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListPoliciesCommand(input, context);
    };
    ListPoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListPoliciesCommand(output, context);
    };
    return ListPoliciesCommand;
}($Command));
export { ListPoliciesCommand };
