// @aws-sdk/client-iam/dist-es/commands/ListSAMLProviderTagsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSAMLProviderTagsRequest, ListSAMLProviderTagsResponse } from "../models/models_0";
import { deserializeAws_queryListSAMLProviderTagsCommand, serializeAws_queryListSAMLProviderTagsCommand, } from "../protocols/Aws_query";
var ListSAMLProviderTagsCommand = (function (_super) {
    __extends(ListSAMLProviderTagsCommand, _super);
    function ListSAMLProviderTagsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSAMLProviderTagsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListSAMLProviderTagsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSAMLProviderTagsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListSAMLProviderTagsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSAMLProviderTagsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListSAMLProviderTagsCommand(input, context);
    };
    ListSAMLProviderTagsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListSAMLProviderTagsCommand(output, context);
    };
    return ListSAMLProviderTagsCommand;
}($Command));
export { ListSAMLProviderTagsCommand };
