// @aws-sdk/client-iam/dist-es/commands/ListAccountAliasesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListAccountAliasesRequest, ListAccountAliasesResponse } from "../models/models_0";
import { deserializeAws_queryListAccountAliasesCommand, serializeAws_queryListAccountAliasesCommand, } from "../protocols/Aws_query";
var ListAccountAliasesCommand = (function (_super) {
    __extends(ListAccountAliasesCommand, _super);
    function ListAccountAliasesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListAccountAliasesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListAccountAliasesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListAccountAliasesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListAccountAliasesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListAccountAliasesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListAccountAliasesCommand(input, context);
    };
    ListAccountAliasesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListAccountAliasesCommand(output, context);
    };
    return ListAccountAliasesCommand;
}($Command));
export { ListAccountAliasesCommand };
