// @aws-sdk/client-iam/dist-es/commands/ListAccessKeysCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListAccessKeysRequest, ListAccessKeysResponse } from "../models/models_0";
import { deserializeAws_queryListAccessKeysCommand, serializeAws_queryListAccessKeysCommand, } from "../protocols/Aws_query";
var ListAccessKeysCommand = (function (_super) {
    __extends(ListAccessKeysCommand, _super);
    function ListAccessKeysCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListAccessKeysCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListAccessKeysCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListAccessKeysRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListAccessKeysResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListAccessKeysCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListAccessKeysCommand(input, context);
    };
    ListAccessKeysCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListAccessKeysCommand(output, context);
    };
    return ListAccessKeysCommand;
}($Command));
export { ListAccessKeysCommand };
