// @aws-sdk/client-iam/dist-es/commands/ListServerCertificatesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListServerCertificatesRequest, ListServerCertificatesResponse } from "../models/models_0";
import { deserializeAws_queryListServerCertificatesCommand, serializeAws_queryListServerCertificatesCommand, } from "../protocols/Aws_query";
var ListServerCertificatesCommand = (function (_super) {
    __extends(ListServerCertificatesCommand, _super);
    function ListServerCertificatesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListServerCertificatesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ListServerCertificatesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListServerCertificatesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListServerCertificatesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListServerCertificatesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListServerCertificatesCommand(input, context);
    };
    ListServerCertificatesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListServerCertificatesCommand(output, context);
    };
    return ListServerCertificatesCommand;
}($Command));
export { ListServerCertificatesCommand };
