// @aws-sdk/client-iam/dist-es/commands/PutRolePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutRolePolicyRequest } from "../models/models_0";
import { deserializeAws_queryPutRolePolicyCommand, serializeAws_queryPutRolePolicyCommand, } from "../protocols/Aws_query";
var PutRolePolicyCommand = (function (_super) {
    __extends(PutRolePolicyCommand, _super);
    function PutRolePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutRolePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "PutRolePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutRolePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutRolePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPutRolePolicyCommand(input, context);
    };
    PutRolePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPutRolePolicyCommand(output, context);
    };
    return PutRolePolicyCommand;
}($Command));
export { PutRolePolicyCommand };
