// @aws-sdk/client-iam/dist-es/commands/RemoveClientIDFromOpenIDConnectProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RemoveClientIDFromOpenIDConnectProviderRequest } from "../models/models_0";
import { deserializeAws_queryRemoveClientIDFromOpenIDConnectProviderCommand, serializeAws_queryRemoveClientIDFromOpenIDConnectProviderCommand, } from "../protocols/Aws_query";
var RemoveClientIDFromOpenIDConnectProviderCommand = (function (_super) {
    __extends(RemoveClientIDFromOpenIDConnectProviderCommand, _super);
    function RemoveClientIDFromOpenIDConnectProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RemoveClientIDFromOpenIDConnectProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "RemoveClientIDFromOpenIDConnectProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RemoveClientIDFromOpenIDConnectProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RemoveClientIDFromOpenIDConnectProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRemoveClientIDFromOpenIDConnectProviderCommand(input, context);
    };
    RemoveClientIDFromOpenIDConnectProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRemoveClientIDFromOpenIDConnectProviderCommand(output, context);
    };
    return RemoveClientIDFromOpenIDConnectProviderCommand;
}($Command));
export { RemoveClientIDFromOpenIDConnectProviderCommand };
