// @aws-sdk/client-iam/dist-es/commands/RemoveUserFromGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RemoveUserFromGroupRequest } from "../models/models_0";
import { deserializeAws_queryRemoveUserFromGroupCommand, serializeAws_queryRemoveUserFromGroupCommand, } from "../protocols/Aws_query";
var RemoveUserFromGroupCommand = (function (_super) {
    __extends(RemoveUserFromGroupCommand, _super);
    function RemoveUserFromGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RemoveUserFromGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "RemoveUserFromGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RemoveUserFromGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RemoveUserFromGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRemoveUserFromGroupCommand(input, context);
    };
    RemoveUserFromGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRemoveUserFromGroupCommand(output, context);
    };
    return RemoveUserFromGroupCommand;
}($Command));
export { RemoveUserFromGroupCommand };
