// @aws-sdk/client-iam/dist-es/commands/ResetServiceSpecificCredentialCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ResetServiceSpecificCredentialRequest, ResetServiceSpecificCredentialResponse } from "../models/models_0";
import { deserializeAws_queryResetServiceSpecificCredentialCommand, serializeAws_queryResetServiceSpecificCredentialCommand, } from "../protocols/Aws_query";
var ResetServiceSpecificCredentialCommand = (function (_super) {
    __extends(ResetServiceSpecificCredentialCommand, _super);
    function ResetServiceSpecificCredentialCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ResetServiceSpecificCredentialCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "ResetServiceSpecificCredentialCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResetServiceSpecificCredentialRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ResetServiceSpecificCredentialResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResetServiceSpecificCredentialCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryResetServiceSpecificCredentialCommand(input, context);
    };
    ResetServiceSpecificCredentialCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryResetServiceSpecificCredentialCommand(output, context);
    };
    return ResetServiceSpecificCredentialCommand;
}($Command));
export { ResetServiceSpecificCredentialCommand };
