// @aws-sdk/client-iam/dist-es/commands/SetDefaultPolicyVersionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetDefaultPolicyVersionRequest } from "../models/models_0";
import { deserializeAws_querySetDefaultPolicyVersionCommand, serializeAws_querySetDefaultPolicyVersionCommand, } from "../protocols/Aws_query";
var SetDefaultPolicyVersionCommand = (function (_super) {
    __extends(SetDefaultPolicyVersionCommand, _super);
    function SetDefaultPolicyVersionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetDefaultPolicyVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "SetDefaultPolicyVersionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetDefaultPolicyVersionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetDefaultPolicyVersionCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetDefaultPolicyVersionCommand(input, context);
    };
    SetDefaultPolicyVersionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetDefaultPolicyVersionCommand(output, context);
    };
    return SetDefaultPolicyVersionCommand;
}($Command));
export { SetDefaultPolicyVersionCommand };
