// @aws-sdk/client-iam/dist-es/commands/SimulateCustomPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SimulateCustomPolicyRequest, SimulatePolicyResponse } from "../models/models_0";
import { deserializeAws_querySimulateCustomPolicyCommand, serializeAws_querySimulateCustomPolicyCommand, } from "../protocols/Aws_query";
var SimulateCustomPolicyCommand = (function (_super) {
    __extends(SimulateCustomPolicyCommand, _super);
    function SimulateCustomPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SimulateCustomPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "SimulateCustomPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SimulateCustomPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: SimulatePolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SimulateCustomPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySimulateCustomPolicyCommand(input, context);
    };
    SimulateCustomPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySimulateCustomPolicyCommand(output, context);
    };
    return SimulateCustomPolicyCommand;
}($Command));
export { SimulateCustomPolicyCommand };
