// @aws-sdk/client-iam/dist-es/commands/TagMFADeviceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TagMFADeviceRequest } from "../models/models_0";
import { deserializeAws_queryTagMFADeviceCommand, serializeAws_queryTagMFADeviceCommand } from "../protocols/Aws_query";
var TagMFADeviceCommand = (function (_super) {
    __extends(TagMFADeviceCommand, _super);
    function TagMFADeviceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TagMFADeviceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "TagMFADeviceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TagMFADeviceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TagMFADeviceCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryTagMFADeviceCommand(input, context);
    };
    TagMFADeviceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryTagMFADeviceCommand(output, context);
    };
    return TagMFADeviceCommand;
}($Command));
export { TagMFADeviceCommand };
