// @aws-sdk/client-iam/dist-es/commands/TagOpenIDConnectProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TagOpenIDConnectProviderRequest } from "../models/models_0";
import { deserializeAws_queryTagOpenIDConnectProviderCommand, serializeAws_queryTagOpenIDConnectProviderCommand, } from "../protocols/Aws_query";
var TagOpenIDConnectProviderCommand = (function (_super) {
    __extends(TagOpenIDConnectProviderCommand, _super);
    function TagOpenIDConnectProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TagOpenIDConnectProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "TagOpenIDConnectProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TagOpenIDConnectProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TagOpenIDConnectProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryTagOpenIDConnectProviderCommand(input, context);
    };
    TagOpenIDConnectProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryTagOpenIDConnectProviderCommand(output, context);
    };
    return TagOpenIDConnectProviderCommand;
}($Command));
export { TagOpenIDConnectProviderCommand };
