// @aws-sdk/client-iam/dist-es/commands/TagPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TagPolicyRequest } from "../models/models_0";
import { deserializeAws_queryTagPolicyCommand, serializeAws_queryTagPolicyCommand } from "../protocols/Aws_query";
var TagPolicyCommand = (function (_super) {
    __extends(TagPolicyCommand, _super);
    function TagPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TagPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "TagPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TagPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TagPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryTagPolicyCommand(input, context);
    };
    TagPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryTagPolicyCommand(output, context);
    };
    return TagPolicyCommand;
}($Command));
export { TagPolicyCommand };
