// @aws-sdk/client-iam/dist-es/commands/TagRoleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TagRoleRequest } from "../models/models_0";
import { deserializeAws_queryTagRoleCommand, serializeAws_queryTagRoleCommand } from "../protocols/Aws_query";
var TagRoleCommand = (function (_super) {
    __extends(TagRoleCommand, _super);
    function TagRoleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TagRoleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "TagRoleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TagRoleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TagRoleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryTagRoleCommand(input, context);
    };
    TagRoleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryTagRoleCommand(output, context);
    };
    return TagRoleCommand;
}($Command));
export { TagRoleCommand };
