// @aws-sdk/client-iam/dist-es/commands/TagSAMLProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TagSAMLProviderRequest } from "../models/models_0";
import { deserializeAws_queryTagSAMLProviderCommand, serializeAws_queryTagSAMLProviderCommand, } from "../protocols/Aws_query";
var TagSAMLProviderCommand = (function (_super) {
    __extends(TagSAMLProviderCommand, _super);
    function TagSAMLProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TagSAMLProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "TagSAMLProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TagSAMLProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TagSAMLProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryTagSAMLProviderCommand(input, context);
    };
    TagSAMLProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryTagSAMLProviderCommand(output, context);
    };
    return TagSAMLProviderCommand;
}($Command));
export { TagSAMLProviderCommand };
