// @aws-sdk/client-iam/dist-es/commands/UntagSAMLProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UntagSAMLProviderRequest } from "../models/models_1";
import { deserializeAws_queryUntagSAMLProviderCommand, serializeAws_queryUntagSAMLProviderCommand, } from "../protocols/Aws_query";
var UntagSAMLProviderCommand = (function (_super) {
    __extends(UntagSAMLProviderCommand, _super);
    function UntagSAMLProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UntagSAMLProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UntagSAMLProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UntagSAMLProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UntagSAMLProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUntagSAMLProviderCommand(input, context);
    };
    UntagSAMLProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUntagSAMLProviderCommand(output, context);
    };
    return UntagSAMLProviderCommand;
}($Command));
export { UntagSAMLProviderCommand };
