// @aws-sdk/client-iam/dist-es/commands/UntagServerCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UntagServerCertificateRequest } from "../models/models_1";
import { deserializeAws_queryUntagServerCertificateCommand, serializeAws_queryUntagServerCertificateCommand, } from "../protocols/Aws_query";
var UntagServerCertificateCommand = (function (_super) {
    __extends(UntagServerCertificateCommand, _super);
    function UntagServerCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UntagServerCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UntagServerCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UntagServerCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UntagServerCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUntagServerCertificateCommand(input, context);
    };
    UntagServerCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUntagServerCertificateCommand(output, context);
    };
    return UntagServerCertificateCommand;
}($Command));
export { UntagServerCertificateCommand };
