// @aws-sdk/client-iam/dist-es/commands/UntagRoleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UntagRoleRequest } from "../models/models_1";
import { deserializeAws_queryUntagRoleCommand, serializeAws_queryUntagRoleCommand } from "../protocols/Aws_query";
var UntagRoleCommand = (function (_super) {
    __extends(UntagRoleCommand, _super);
    function UntagRoleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UntagRoleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UntagRoleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UntagRoleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UntagRoleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUntagRoleCommand(input, context);
    };
    UntagRoleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUntagRoleCommand(output, context);
    };
    return UntagRoleCommand;
}($Command));
export { UntagRoleCommand };
