// @aws-sdk/client-iam/dist-es/commands/UntagOpenIDConnectProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UntagOpenIDConnectProviderRequest } from "../models/models_0";
import { deserializeAws_queryUntagOpenIDConnectProviderCommand, serializeAws_queryUntagOpenIDConnectProviderCommand, } from "../protocols/Aws_query";
var UntagOpenIDConnectProviderCommand = (function (_super) {
    __extends(UntagOpenIDConnectProviderCommand, _super);
    function UntagOpenIDConnectProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UntagOpenIDConnectProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UntagOpenIDConnectProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UntagOpenIDConnectProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UntagOpenIDConnectProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUntagOpenIDConnectProviderCommand(input, context);
    };
    UntagOpenIDConnectProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUntagOpenIDConnectProviderCommand(output, context);
    };
    return UntagOpenIDConnectProviderCommand;
}($Command));
export { UntagOpenIDConnectProviderCommand };
