// @aws-sdk/client-iam/dist-es/commands/UntagInstanceProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UntagInstanceProfileRequest } from "../models/models_0";
import { deserializeAws_queryUntagInstanceProfileCommand, serializeAws_queryUntagInstanceProfileCommand, } from "../protocols/Aws_query";
var UntagInstanceProfileCommand = (function (_super) {
    __extends(UntagInstanceProfileCommand, _super);
    function UntagInstanceProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UntagInstanceProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UntagInstanceProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UntagInstanceProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UntagInstanceProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUntagInstanceProfileCommand(input, context);
    };
    UntagInstanceProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUntagInstanceProfileCommand(output, context);
    };
    return UntagInstanceProfileCommand;
}($Command));
export { UntagInstanceProfileCommand };
