// @aws-sdk/client-iam/dist-es/commands/UpdateServiceSpecificCredentialCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateServiceSpecificCredentialRequest } from "../models/models_1";
import { deserializeAws_queryUpdateServiceSpecificCredentialCommand, serializeAws_queryUpdateServiceSpecificCredentialCommand, } from "../protocols/Aws_query";
var UpdateServiceSpecificCredentialCommand = (function (_super) {
    __extends(UpdateServiceSpecificCredentialCommand, _super);
    function UpdateServiceSpecificCredentialCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateServiceSpecificCredentialCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UpdateServiceSpecificCredentialCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateServiceSpecificCredentialRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateServiceSpecificCredentialCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateServiceSpecificCredentialCommand(input, context);
    };
    UpdateServiceSpecificCredentialCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateServiceSpecificCredentialCommand(output, context);
    };
    return UpdateServiceSpecificCredentialCommand;
}($Command));
export { UpdateServiceSpecificCredentialCommand };
