// @aws-sdk/client-iam/dist-es/commands/UpdateUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUserRequest } from "../models/models_1";
import { deserializeAws_queryUpdateUserCommand, serializeAws_queryUpdateUserCommand } from "../protocols/Aws_query";
var UpdateUserCommand = (function (_super) {
    __extends(UpdateUserCommand, _super);
    function UpdateUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UpdateUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateUserCommand(input, context);
    };
    UpdateUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateUserCommand(output, context);
    };
    return UpdateUserCommand;
}($Command));
export { UpdateUserCommand };
