// @aws-sdk/client-iam/dist-es/commands/UpdateAccessKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateAccessKeyRequest } from "../models/models_1";
import { deserializeAws_queryUpdateAccessKeyCommand, serializeAws_queryUpdateAccessKeyCommand, } from "../protocols/Aws_query";
var UpdateAccessKeyCommand = (function (_super) {
    __extends(UpdateAccessKeyCommand, _super);
    function UpdateAccessKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateAccessKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UpdateAccessKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateAccessKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateAccessKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateAccessKeyCommand(input, context);
    };
    UpdateAccessKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateAccessKeyCommand(output, context);
    };
    return UpdateAccessKeyCommand;
}($Command));
export { UpdateAccessKeyCommand };
