// @aws-sdk/client-iam/dist-es/commands/UpdateAccountPasswordPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateAccountPasswordPolicyRequest } from "../models/models_1";
import { deserializeAws_queryUpdateAccountPasswordPolicyCommand, serializeAws_queryUpdateAccountPasswordPolicyCommand, } from "../protocols/Aws_query";
var UpdateAccountPasswordPolicyCommand = (function (_super) {
    __extends(UpdateAccountPasswordPolicyCommand, _super);
    function UpdateAccountPasswordPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateAccountPasswordPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UpdateAccountPasswordPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateAccountPasswordPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateAccountPasswordPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateAccountPasswordPolicyCommand(input, context);
    };
    UpdateAccountPasswordPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateAccountPasswordPolicyCommand(output, context);
    };
    return UpdateAccountPasswordPolicyCommand;
}($Command));
export { UpdateAccountPasswordPolicyCommand };
