// @aws-sdk/client-iam/dist-es/commands/UpdateSSHPublicKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateSSHPublicKeyRequest } from "../models/models_1";
import { deserializeAws_queryUpdateSSHPublicKeyCommand, serializeAws_queryUpdateSSHPublicKeyCommand, } from "../protocols/Aws_query";
var UpdateSSHPublicKeyCommand = (function (_super) {
    __extends(UpdateSSHPublicKeyCommand, _super);
    function UpdateSSHPublicKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateSSHPublicKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UpdateSSHPublicKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateSSHPublicKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateSSHPublicKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateSSHPublicKeyCommand(input, context);
    };
    UpdateSSHPublicKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateSSHPublicKeyCommand(output, context);
    };
    return UpdateSSHPublicKeyCommand;
}($Command));
export { UpdateSSHPublicKeyCommand };
