// @aws-sdk/client-iam/dist-es/commands/UpdateRoleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateRoleRequest, UpdateRoleResponse } from "../models/models_1";
import { deserializeAws_queryUpdateRoleCommand, serializeAws_queryUpdateRoleCommand } from "../protocols/Aws_query";
var UpdateRoleCommand = (function (_super) {
    __extends(UpdateRoleCommand, _super);
    function UpdateRoleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateRoleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UpdateRoleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateRoleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateRoleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateRoleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateRoleCommand(input, context);
    };
    UpdateRoleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateRoleCommand(output, context);
    };
    return UpdateRoleCommand;
}($Command));
export { UpdateRoleCommand };
