// @aws-sdk/client-iam/dist-es/commands/UploadSigningCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UploadSigningCertificateRequest, UploadSigningCertificateResponse } from "../models/models_1";
import { deserializeAws_queryUploadSigningCertificateCommand, serializeAws_queryUploadSigningCertificateCommand, } from "../protocols/Aws_query";
var UploadSigningCertificateCommand = (function (_super) {
    __extends(UploadSigningCertificateCommand, _super);
    function UploadSigningCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UploadSigningCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UploadSigningCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UploadSigningCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UploadSigningCertificateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UploadSigningCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUploadSigningCertificateCommand(input, context);
    };
    UploadSigningCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUploadSigningCertificateCommand(output, context);
    };
    return UploadSigningCertificateCommand;
}($Command));
export { UploadSigningCertificateCommand };
