// @aws-sdk/client-iam/dist-es/commands/UploadSSHPublicKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UploadSSHPublicKeyRequest, UploadSSHPublicKeyResponse } from "../models/models_1";
import { deserializeAws_queryUploadSSHPublicKeyCommand, serializeAws_queryUploadSSHPublicKeyCommand, } from "../protocols/Aws_query";
var UploadSSHPublicKeyCommand = (function (_super) {
    __extends(UploadSSHPublicKeyCommand, _super);
    function UploadSSHPublicKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UploadSSHPublicKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "UploadSSHPublicKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UploadSSHPublicKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UploadSSHPublicKeyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UploadSSHPublicKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUploadSSHPublicKeyCommand(input, context);
    };
    UploadSSHPublicKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUploadSSHPublicKeyCommand(output, context);
    };
    return UploadSSHPublicKeyCommand;
}($Command));
export { UploadSSHPublicKeyCommand };
