// @aws-sdk/client-iam/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePolicyVersionResponse = exports.PolicyVersion = exports.CreatePolicyVersionRequest = exports.MalformedPolicyDocumentException = exports.CreatePolicyResponse = exports.Policy = exports.CreatePolicyRequest = exports.CreateOpenIDConnectProviderResponse = exports.CreateOpenIDConnectProviderRequest = exports.CreateLoginProfileResponse = exports.LoginProfile = exports.CreateLoginProfileRequest = exports.CreateInstanceProfileResponse = exports.InstanceProfile = exports.Role = exports.RoleLastUsed = exports.CreateInstanceProfileRequest = exports.Tag = exports.ConcurrentModificationException = exports.CreateGroupResponse = exports.Group = exports.CreateGroupRequest = exports.CreateAccountAliasRequest = exports.CreateAccessKeyResponse = exports.CreateAccessKeyRequest = exports.PasswordPolicyViolationException = exports.InvalidUserTypeException = exports.EntityTemporarilyUnmodifiableException = exports.ChangePasswordRequest = exports.AttachUserPolicyRequest = exports.AttachRolePolicyRequest = exports.PolicyNotAttachableException = exports.AttachGroupPolicyRequest = exports.AttachedPolicy = exports.AttachedPermissionsBoundary = exports.PermissionsBoundaryAttachmentType = exports.AddUserToGroupRequest = exports.UnmodifiableEntityException = exports.EntityAlreadyExistsException = exports.AddRoleToInstanceProfileRequest = exports.ServiceFailureException = exports.NoSuchEntityException = exports.LimitExceededException = exports.InvalidInputException = exports.AddClientIDToOpenIDConnectProviderRequest = exports.AccessKeyMetadata = exports.AccessKeyLastUsed = exports.AccessKey = exports.AccessDetail = exports.AccessAdvisorUsageGranularityType = void 0;
exports.GenerateOrganizationsAccessReportResponse = exports.GenerateOrganizationsAccessReportRequest = exports.GenerateCredentialReportResponse = exports.ReportStateType = exports.InvalidAuthenticationCodeException = exports.EnableMFADeviceRequest = exports.DetachUserPolicyRequest = exports.DetachRolePolicyRequest = exports.DetachGroupPolicyRequest = exports.DeleteVirtualMFADeviceRequest = exports.DeleteUserPolicyRequest = exports.DeleteUserPermissionsBoundaryRequest = exports.DeleteUserRequest = exports.DeleteSSHPublicKeyRequest = exports.DeleteSigningCertificateRequest = exports.DeleteServiceSpecificCredentialRequest = exports.DeleteServiceLinkedRoleResponse = exports.DeleteServiceLinkedRoleRequest = exports.DeleteServerCertificateRequest = exports.DeleteSAMLProviderRequest = exports.DeleteRolePolicyRequest = exports.DeleteRolePermissionsBoundaryRequest = exports.DeleteRoleRequest = exports.DeletePolicyVersionRequest = exports.DeletePolicyRequest = exports.DeleteOpenIDConnectProviderRequest = exports.DeleteLoginProfileRequest = exports.DeleteInstanceProfileRequest = exports.DeleteGroupPolicyRequest = exports.DeleteGroupRequest = exports.DeleteConflictException = exports.DeleteAccountAliasRequest = exports.DeleteAccessKeyRequest = exports.DeactivateMFADeviceRequest = exports.CreateVirtualMFADeviceResponse = exports.VirtualMFADevice = exports.CreateVirtualMFADeviceRequest = exports.CreateUserResponse = exports.User = exports.CreateUserRequest = exports.ServiceNotSupportedException = exports.CreateServiceSpecificCredentialResponse = exports.ServiceSpecificCredential = exports.CreateServiceSpecificCredentialRequest = exports.CreateServiceLinkedRoleResponse = exports.CreateServiceLinkedRoleRequest = exports.CreateSAMLProviderResponse = exports.CreateSAMLProviderRequest = exports.CreateRoleResponse = exports.CreateRoleRequest = void 0;
exports.ServerCertificate = exports.ServerCertificateMetadata = exports.GetServerCertificateRequest = exports.GetSAMLProviderResponse = exports.GetSAMLProviderRequest = exports.GetRolePolicyResponse = exports.GetRolePolicyRequest = exports.GetRoleResponse = exports.GetRoleRequest = exports.GetPolicyVersionResponse = exports.GetPolicyVersionRequest = exports.GetPolicyResponse = exports.GetPolicyRequest = exports.GetOrganizationsAccessReportResponse = exports.ErrorDetails = exports.GetOrganizationsAccessReportRequest = exports.SortKeyType = exports.GetOpenIDConnectProviderResponse = exports.GetOpenIDConnectProviderRequest = exports.GetLoginProfileResponse = exports.GetLoginProfileRequest = exports.GetInstanceProfileResponse = exports.GetInstanceProfileRequest = exports.GetGroupPolicyResponse = exports.GetGroupPolicyRequest = exports.GetGroupResponse = exports.GetGroupRequest = exports.GetCredentialReportResponse = exports.CredentialReportNotReadyException = exports.CredentialReportNotPresentException = exports.CredentialReportExpiredException = exports.GetContextKeysForPrincipalPolicyRequest = exports.GetContextKeysForPolicyResponse = exports.GetContextKeysForCustomPolicyRequest = exports.GetAccountSummaryResponse = exports.GetAccountPasswordPolicyResponse = exports.PasswordPolicy = exports.GetAccountAuthorizationDetailsResponse = exports.UserDetail = exports.RoleDetail = exports.ManagedPolicyDetail = exports.GroupDetail = exports.PolicyDetail = exports.GetAccountAuthorizationDetailsRequest = exports.EntityType = exports.GetAccessKeyLastUsedResponse = exports.GetAccessKeyLastUsedRequest = exports.GenerateServiceLastAccessedDetailsResponse = exports.GenerateServiceLastAccessedDetailsRequest = exports.ReportGenerationLimitExceededException = void 0;
exports.ListInstanceProfileTagsResponse = exports.ListInstanceProfileTagsRequest = exports.ListInstanceProfilesForRoleResponse = exports.ListInstanceProfilesForRoleRequest = exports.ListInstanceProfilesResponse = exports.ListInstanceProfilesRequest = exports.ListGroupsForUserResponse = exports.ListGroupsForUserRequest = exports.ListGroupsResponse = exports.ListGroupsRequest = exports.ListGroupPoliciesResponse = exports.ListGroupPoliciesRequest = exports.ListEntitiesForPolicyResponse = exports.PolicyUser = exports.PolicyRole = exports.PolicyGroup = exports.ListEntitiesForPolicyRequest = exports.ListAttachedUserPoliciesResponse = exports.ListAttachedUserPoliciesRequest = exports.ListAttachedRolePoliciesResponse = exports.ListAttachedRolePoliciesRequest = exports.ListAttachedGroupPoliciesResponse = exports.ListAttachedGroupPoliciesRequest = exports.ListAccountAliasesResponse = exports.ListAccountAliasesRequest = exports.ListAccessKeysResponse = exports.ListAccessKeysRequest = exports.GetUserPolicyResponse = exports.GetUserPolicyRequest = exports.GetUserResponse = exports.GetUserRequest = exports.UnrecognizedPublicKeyEncodingException = exports.GetSSHPublicKeyResponse = exports.SSHPublicKey = exports.GetSSHPublicKeyRequest = exports.GetServiceLinkedRoleDeletionStatusResponse = exports.DeletionTaskStatusType = exports.DeletionTaskFailureReasonType = exports.RoleUsageType = exports.GetServiceLinkedRoleDeletionStatusRequest = exports.GetServiceLastAccessedDetailsWithEntitiesResponse = exports.EntityDetails = exports.EntityInfo = exports.PolicyOwnerEntityType = exports.GetServiceLastAccessedDetailsWithEntitiesRequest = exports.GetServiceLastAccessedDetailsResponse = exports.ServiceLastAccessed = exports.TrackedActionLastAccessed = exports.GetServiceLastAccessedDetailsRequest = exports.GetServerCertificateResponse = void 0;
exports.ListUsersResponse = exports.ListUsersRequest = exports.ListUserPoliciesResponse = exports.ListUserPoliciesRequest = exports.ListSSHPublicKeysResponse = exports.SSHPublicKeyMetadata = exports.ListSSHPublicKeysRequest = exports.ListSigningCertificatesResponse = exports.SigningCertificate = exports.ListSigningCertificatesRequest = exports.ListServiceSpecificCredentialsResponse = exports.ServiceSpecificCredentialMetadata = exports.ListServiceSpecificCredentialsRequest = exports.ListServerCertificateTagsResponse = exports.ListServerCertificateTagsRequest = exports.ListServerCertificatesResponse = exports.ListServerCertificatesRequest = exports.ListSAMLProviderTagsResponse = exports.ListSAMLProviderTagsRequest = exports.ListSAMLProvidersResponse = exports.SAMLProviderListEntry = exports.ListSAMLProvidersRequest = exports.ListRoleTagsResponse = exports.ListRoleTagsRequest = exports.ListRolesResponse = exports.ListRolesRequest = exports.ListRolePoliciesResponse = exports.ListRolePoliciesRequest = exports.ListPolicyVersionsResponse = exports.ListPolicyVersionsRequest = exports.ListPolicyTagsResponse = exports.ListPolicyTagsRequest = exports.ListPoliciesGrantingServiceAccessResponse = exports.ListPoliciesGrantingServiceAccessEntry = exports.PolicyGrantingServiceAccess = exports.PolicyType = exports.ListPoliciesGrantingServiceAccessRequest = exports.ListPoliciesResponse = exports.ListPoliciesRequest = exports.PolicyScopeType = exports.ListOpenIDConnectProviderTagsResponse = exports.ListOpenIDConnectProviderTagsRequest = exports.ListOpenIDConnectProvidersResponse = exports.OpenIDConnectProviderListEntry = exports.ListOpenIDConnectProvidersRequest = exports.ListMFADeviceTagsResponse = exports.ListMFADeviceTagsRequest = exports.ListMFADevicesResponse = exports.MFADevice = exports.ListMFADevicesRequest = void 0;
exports.UntagOpenIDConnectProviderRequest = exports.UntagMFADeviceRequest = exports.UntagInstanceProfileRequest = exports.TagUserRequest = exports.TagServerCertificateRequest = exports.TagSAMLProviderRequest = exports.TagRoleRequest = exports.TagPolicyRequest = exports.TagOpenIDConnectProviderRequest = exports.TagMFADeviceRequest = exports.TagInstanceProfileRequest = exports.SimulatePrincipalPolicyRequest = exports.SimulatePolicyResponse = exports.EvaluationResult = exports.ResourceSpecificResult = exports.PermissionsBoundaryDecisionDetail = exports.OrganizationsDecisionDetail = exports.Statement = exports.PolicySourceType = exports.Position = exports.PolicyEvaluationDecisionType = exports.SimulateCustomPolicyRequest = exports.ContextEntry = exports.ContextKeyTypeEnum = exports.PolicyEvaluationException = exports.SetSecurityTokenServicePreferencesRequest = exports.GlobalEndpointTokenVersion = exports.SetDefaultPolicyVersionRequest = exports.ResyncMFADeviceRequest = exports.ResetServiceSpecificCredentialResponse = exports.ResetServiceSpecificCredentialRequest = exports.RemoveUserFromGroupRequest = exports.RemoveRoleFromInstanceProfileRequest = exports.RemoveClientIDFromOpenIDConnectProviderRequest = exports.PutUserPolicyRequest = exports.PutUserPermissionsBoundaryRequest = exports.PutRolePolicyRequest = exports.PutRolePermissionsBoundaryRequest = exports.PutGroupPolicyRequest = exports.ListVirtualMFADevicesResponse = exports.ListVirtualMFADevicesRequest = exports.ListUserTagsResponse = exports.ListUserTagsRequest = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
var AccessAdvisorUsageGranularityType;
(function (AccessAdvisorUsageGranularityType) {
    AccessAdvisorUsageGranularityType["ACTION_LEVEL"] = "ACTION_LEVEL";
    AccessAdvisorUsageGranularityType["SERVICE_LEVEL"] = "SERVICE_LEVEL";
})(AccessAdvisorUsageGranularityType = exports.AccessAdvisorUsageGranularityType || (exports.AccessAdvisorUsageGranularityType = {}));
var AccessDetail;
(function (AccessDetail) {
    AccessDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccessDetail = exports.AccessDetail || (exports.AccessDetail = {}));
var AccessKey;
(function (AccessKey) {
    AccessKey.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.SecretAccessKey && { SecretAccessKey: smithy_client_1.SENSITIVE_STRING }),
    });
})(AccessKey = exports.AccessKey || (exports.AccessKey = {}));
var AccessKeyLastUsed;
(function (AccessKeyLastUsed) {
    AccessKeyLastUsed.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccessKeyLastUsed = exports.AccessKeyLastUsed || (exports.AccessKeyLastUsed = {}));
var AccessKeyMetadata;
(function (AccessKeyMetadata) {
    AccessKeyMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccessKeyMetadata = exports.AccessKeyMetadata || (exports.AccessKeyMetadata = {}));
var AddClientIDToOpenIDConnectProviderRequest;
(function (AddClientIDToOpenIDConnectProviderRequest) {
    AddClientIDToOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddClientIDToOpenIDConnectProviderRequest = exports.AddClientIDToOpenIDConnectProviderRequest || (exports.AddClientIDToOpenIDConnectProviderRequest = {}));
var InvalidInputException;
(function (InvalidInputException) {
    InvalidInputException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidInputException = exports.InvalidInputException || (exports.InvalidInputException = {}));
var LimitExceededException;
(function (LimitExceededException) {
    LimitExceededException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LimitExceededException = exports.LimitExceededException || (exports.LimitExceededException = {}));
var NoSuchEntityException;
(function (NoSuchEntityException) {
    NoSuchEntityException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(NoSuchEntityException = exports.NoSuchEntityException || (exports.NoSuchEntityException = {}));
var ServiceFailureException;
(function (ServiceFailureException) {
    ServiceFailureException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServiceFailureException = exports.ServiceFailureException || (exports.ServiceFailureException = {}));
var AddRoleToInstanceProfileRequest;
(function (AddRoleToInstanceProfileRequest) {
    AddRoleToInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddRoleToInstanceProfileRequest = exports.AddRoleToInstanceProfileRequest || (exports.AddRoleToInstanceProfileRequest = {}));
var EntityAlreadyExistsException;
(function (EntityAlreadyExistsException) {
    EntityAlreadyExistsException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EntityAlreadyExistsException = exports.EntityAlreadyExistsException || (exports.EntityAlreadyExistsException = {}));
var UnmodifiableEntityException;
(function (UnmodifiableEntityException) {
    UnmodifiableEntityException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UnmodifiableEntityException = exports.UnmodifiableEntityException || (exports.UnmodifiableEntityException = {}));
var AddUserToGroupRequest;
(function (AddUserToGroupRequest) {
    AddUserToGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddUserToGroupRequest = exports.AddUserToGroupRequest || (exports.AddUserToGroupRequest = {}));
var PermissionsBoundaryAttachmentType;
(function (PermissionsBoundaryAttachmentType) {
    PermissionsBoundaryAttachmentType["Policy"] = "PermissionsBoundaryPolicy";
})(PermissionsBoundaryAttachmentType = exports.PermissionsBoundaryAttachmentType || (exports.PermissionsBoundaryAttachmentType = {}));
var AttachedPermissionsBoundary;
(function (AttachedPermissionsBoundary) {
    AttachedPermissionsBoundary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AttachedPermissionsBoundary = exports.AttachedPermissionsBoundary || (exports.AttachedPermissionsBoundary = {}));
var AttachedPolicy;
(function (AttachedPolicy) {
    AttachedPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AttachedPolicy = exports.AttachedPolicy || (exports.AttachedPolicy = {}));
var AttachGroupPolicyRequest;
(function (AttachGroupPolicyRequest) {
    AttachGroupPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AttachGroupPolicyRequest = exports.AttachGroupPolicyRequest || (exports.AttachGroupPolicyRequest = {}));
var PolicyNotAttachableException;
(function (PolicyNotAttachableException) {
    PolicyNotAttachableException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyNotAttachableException = exports.PolicyNotAttachableException || (exports.PolicyNotAttachableException = {}));
var AttachRolePolicyRequest;
(function (AttachRolePolicyRequest) {
    AttachRolePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AttachRolePolicyRequest = exports.AttachRolePolicyRequest || (exports.AttachRolePolicyRequest = {}));
var AttachUserPolicyRequest;
(function (AttachUserPolicyRequest) {
    AttachUserPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AttachUserPolicyRequest = exports.AttachUserPolicyRequest || (exports.AttachUserPolicyRequest = {}));
var ChangePasswordRequest;
(function (ChangePasswordRequest) {
    ChangePasswordRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.OldPassword && { OldPassword: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.NewPassword && { NewPassword: smithy_client_1.SENSITIVE_STRING }),
    });
})(ChangePasswordRequest = exports.ChangePasswordRequest || (exports.ChangePasswordRequest = {}));
var EntityTemporarilyUnmodifiableException;
(function (EntityTemporarilyUnmodifiableException) {
    EntityTemporarilyUnmodifiableException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EntityTemporarilyUnmodifiableException = exports.EntityTemporarilyUnmodifiableException || (exports.EntityTemporarilyUnmodifiableException = {}));
var InvalidUserTypeException;
(function (InvalidUserTypeException) {
    InvalidUserTypeException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidUserTypeException = exports.InvalidUserTypeException || (exports.InvalidUserTypeException = {}));
var PasswordPolicyViolationException;
(function (PasswordPolicyViolationException) {
    PasswordPolicyViolationException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PasswordPolicyViolationException = exports.PasswordPolicyViolationException || (exports.PasswordPolicyViolationException = {}));
var CreateAccessKeyRequest;
(function (CreateAccessKeyRequest) {
    CreateAccessKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateAccessKeyRequest = exports.CreateAccessKeyRequest || (exports.CreateAccessKeyRequest = {}));
var CreateAccessKeyResponse;
(function (CreateAccessKeyResponse) {
    CreateAccessKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.AccessKey && { AccessKey: AccessKey.filterSensitiveLog(obj.AccessKey) }),
    });
})(CreateAccessKeyResponse = exports.CreateAccessKeyResponse || (exports.CreateAccessKeyResponse = {}));
var CreateAccountAliasRequest;
(function (CreateAccountAliasRequest) {
    CreateAccountAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateAccountAliasRequest = exports.CreateAccountAliasRequest || (exports.CreateAccountAliasRequest = {}));
var CreateGroupRequest;
(function (CreateGroupRequest) {
    CreateGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateGroupRequest = exports.CreateGroupRequest || (exports.CreateGroupRequest = {}));
var Group;
(function (Group) {
    Group.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Group = exports.Group || (exports.Group = {}));
var CreateGroupResponse;
(function (CreateGroupResponse) {
    CreateGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateGroupResponse = exports.CreateGroupResponse || (exports.CreateGroupResponse = {}));
var ConcurrentModificationException;
(function (ConcurrentModificationException) {
    ConcurrentModificationException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConcurrentModificationException = exports.ConcurrentModificationException || (exports.ConcurrentModificationException = {}));
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var CreateInstanceProfileRequest;
(function (CreateInstanceProfileRequest) {
    CreateInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateInstanceProfileRequest = exports.CreateInstanceProfileRequest || (exports.CreateInstanceProfileRequest = {}));
var RoleLastUsed;
(function (RoleLastUsed) {
    RoleLastUsed.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RoleLastUsed = exports.RoleLastUsed || (exports.RoleLastUsed = {}));
var Role;
(function (Role) {
    Role.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Role = exports.Role || (exports.Role = {}));
var InstanceProfile;
(function (InstanceProfile) {
    InstanceProfile.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InstanceProfile = exports.InstanceProfile || (exports.InstanceProfile = {}));
var CreateInstanceProfileResponse;
(function (CreateInstanceProfileResponse) {
    CreateInstanceProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateInstanceProfileResponse = exports.CreateInstanceProfileResponse || (exports.CreateInstanceProfileResponse = {}));
var CreateLoginProfileRequest;
(function (CreateLoginProfileRequest) {
    CreateLoginProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Password && { Password: smithy_client_1.SENSITIVE_STRING }),
    });
})(CreateLoginProfileRequest = exports.CreateLoginProfileRequest || (exports.CreateLoginProfileRequest = {}));
var LoginProfile;
(function (LoginProfile) {
    LoginProfile.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LoginProfile = exports.LoginProfile || (exports.LoginProfile = {}));
var CreateLoginProfileResponse;
(function (CreateLoginProfileResponse) {
    CreateLoginProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateLoginProfileResponse = exports.CreateLoginProfileResponse || (exports.CreateLoginProfileResponse = {}));
var CreateOpenIDConnectProviderRequest;
(function (CreateOpenIDConnectProviderRequest) {
    CreateOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateOpenIDConnectProviderRequest = exports.CreateOpenIDConnectProviderRequest || (exports.CreateOpenIDConnectProviderRequest = {}));
var CreateOpenIDConnectProviderResponse;
(function (CreateOpenIDConnectProviderResponse) {
    CreateOpenIDConnectProviderResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateOpenIDConnectProviderResponse = exports.CreateOpenIDConnectProviderResponse || (exports.CreateOpenIDConnectProviderResponse = {}));
var CreatePolicyRequest;
(function (CreatePolicyRequest) {
    CreatePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatePolicyRequest = exports.CreatePolicyRequest || (exports.CreatePolicyRequest = {}));
var Policy;
(function (Policy) {
    Policy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Policy = exports.Policy || (exports.Policy = {}));
var CreatePolicyResponse;
(function (CreatePolicyResponse) {
    CreatePolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatePolicyResponse = exports.CreatePolicyResponse || (exports.CreatePolicyResponse = {}));
var MalformedPolicyDocumentException;
(function (MalformedPolicyDocumentException) {
    MalformedPolicyDocumentException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MalformedPolicyDocumentException = exports.MalformedPolicyDocumentException || (exports.MalformedPolicyDocumentException = {}));
var CreatePolicyVersionRequest;
(function (CreatePolicyVersionRequest) {
    CreatePolicyVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatePolicyVersionRequest = exports.CreatePolicyVersionRequest || (exports.CreatePolicyVersionRequest = {}));
var PolicyVersion;
(function (PolicyVersion) {
    PolicyVersion.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyVersion = exports.PolicyVersion || (exports.PolicyVersion = {}));
var CreatePolicyVersionResponse;
(function (CreatePolicyVersionResponse) {
    CreatePolicyVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatePolicyVersionResponse = exports.CreatePolicyVersionResponse || (exports.CreatePolicyVersionResponse = {}));
var CreateRoleRequest;
(function (CreateRoleRequest) {
    CreateRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateRoleRequest = exports.CreateRoleRequest || (exports.CreateRoleRequest = {}));
var CreateRoleResponse;
(function (CreateRoleResponse) {
    CreateRoleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateRoleResponse = exports.CreateRoleResponse || (exports.CreateRoleResponse = {}));
var CreateSAMLProviderRequest;
(function (CreateSAMLProviderRequest) {
    CreateSAMLProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateSAMLProviderRequest = exports.CreateSAMLProviderRequest || (exports.CreateSAMLProviderRequest = {}));
var CreateSAMLProviderResponse;
(function (CreateSAMLProviderResponse) {
    CreateSAMLProviderResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateSAMLProviderResponse = exports.CreateSAMLProviderResponse || (exports.CreateSAMLProviderResponse = {}));
var CreateServiceLinkedRoleRequest;
(function (CreateServiceLinkedRoleRequest) {
    CreateServiceLinkedRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateServiceLinkedRoleRequest = exports.CreateServiceLinkedRoleRequest || (exports.CreateServiceLinkedRoleRequest = {}));
var CreateServiceLinkedRoleResponse;
(function (CreateServiceLinkedRoleResponse) {
    CreateServiceLinkedRoleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateServiceLinkedRoleResponse = exports.CreateServiceLinkedRoleResponse || (exports.CreateServiceLinkedRoleResponse = {}));
var CreateServiceSpecificCredentialRequest;
(function (CreateServiceSpecificCredentialRequest) {
    CreateServiceSpecificCredentialRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateServiceSpecificCredentialRequest = exports.CreateServiceSpecificCredentialRequest || (exports.CreateServiceSpecificCredentialRequest = {}));
var ServiceSpecificCredential;
(function (ServiceSpecificCredential) {
    ServiceSpecificCredential.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ServicePassword && { ServicePassword: smithy_client_1.SENSITIVE_STRING }),
    });
})(ServiceSpecificCredential = exports.ServiceSpecificCredential || (exports.ServiceSpecificCredential = {}));
var CreateServiceSpecificCredentialResponse;
(function (CreateServiceSpecificCredentialResponse) {
    CreateServiceSpecificCredentialResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ServiceSpecificCredential && {
            ServiceSpecificCredential: ServiceSpecificCredential.filterSensitiveLog(obj.ServiceSpecificCredential),
        }),
    });
})(CreateServiceSpecificCredentialResponse = exports.CreateServiceSpecificCredentialResponse || (exports.CreateServiceSpecificCredentialResponse = {}));
var ServiceNotSupportedException;
(function (ServiceNotSupportedException) {
    ServiceNotSupportedException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServiceNotSupportedException = exports.ServiceNotSupportedException || (exports.ServiceNotSupportedException = {}));
var CreateUserRequest;
(function (CreateUserRequest) {
    CreateUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateUserRequest = exports.CreateUserRequest || (exports.CreateUserRequest = {}));
var User;
(function (User) {
    User.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(User = exports.User || (exports.User = {}));
var CreateUserResponse;
(function (CreateUserResponse) {
    CreateUserResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateUserResponse = exports.CreateUserResponse || (exports.CreateUserResponse = {}));
var CreateVirtualMFADeviceRequest;
(function (CreateVirtualMFADeviceRequest) {
    CreateVirtualMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateVirtualMFADeviceRequest = exports.CreateVirtualMFADeviceRequest || (exports.CreateVirtualMFADeviceRequest = {}));
var VirtualMFADevice;
(function (VirtualMFADevice) {
    VirtualMFADevice.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Base32StringSeed && { Base32StringSeed: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.QRCodePNG && { QRCodePNG: smithy_client_1.SENSITIVE_STRING }),
    });
})(VirtualMFADevice = exports.VirtualMFADevice || (exports.VirtualMFADevice = {}));
var CreateVirtualMFADeviceResponse;
(function (CreateVirtualMFADeviceResponse) {
    CreateVirtualMFADeviceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.VirtualMFADevice && { VirtualMFADevice: VirtualMFADevice.filterSensitiveLog(obj.VirtualMFADevice) }),
    });
})(CreateVirtualMFADeviceResponse = exports.CreateVirtualMFADeviceResponse || (exports.CreateVirtualMFADeviceResponse = {}));
var DeactivateMFADeviceRequest;
(function (DeactivateMFADeviceRequest) {
    DeactivateMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeactivateMFADeviceRequest = exports.DeactivateMFADeviceRequest || (exports.DeactivateMFADeviceRequest = {}));
var DeleteAccessKeyRequest;
(function (DeleteAccessKeyRequest) {
    DeleteAccessKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteAccessKeyRequest = exports.DeleteAccessKeyRequest || (exports.DeleteAccessKeyRequest = {}));
var DeleteAccountAliasRequest;
(function (DeleteAccountAliasRequest) {
    DeleteAccountAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteAccountAliasRequest = exports.DeleteAccountAliasRequest || (exports.DeleteAccountAliasRequest = {}));
var DeleteConflictException;
(function (DeleteConflictException) {
    DeleteConflictException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteConflictException = exports.DeleteConflictException || (exports.DeleteConflictException = {}));
var DeleteGroupRequest;
(function (DeleteGroupRequest) {
    DeleteGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteGroupRequest = exports.DeleteGroupRequest || (exports.DeleteGroupRequest = {}));
var DeleteGroupPolicyRequest;
(function (DeleteGroupPolicyRequest) {
    DeleteGroupPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteGroupPolicyRequest = exports.DeleteGroupPolicyRequest || (exports.DeleteGroupPolicyRequest = {}));
var DeleteInstanceProfileRequest;
(function (DeleteInstanceProfileRequest) {
    DeleteInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteInstanceProfileRequest = exports.DeleteInstanceProfileRequest || (exports.DeleteInstanceProfileRequest = {}));
var DeleteLoginProfileRequest;
(function (DeleteLoginProfileRequest) {
    DeleteLoginProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteLoginProfileRequest = exports.DeleteLoginProfileRequest || (exports.DeleteLoginProfileRequest = {}));
var DeleteOpenIDConnectProviderRequest;
(function (DeleteOpenIDConnectProviderRequest) {
    DeleteOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteOpenIDConnectProviderRequest = exports.DeleteOpenIDConnectProviderRequest || (exports.DeleteOpenIDConnectProviderRequest = {}));
var DeletePolicyRequest;
(function (DeletePolicyRequest) {
    DeletePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeletePolicyRequest = exports.DeletePolicyRequest || (exports.DeletePolicyRequest = {}));
var DeletePolicyVersionRequest;
(function (DeletePolicyVersionRequest) {
    DeletePolicyVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeletePolicyVersionRequest = exports.DeletePolicyVersionRequest || (exports.DeletePolicyVersionRequest = {}));
var DeleteRoleRequest;
(function (DeleteRoleRequest) {
    DeleteRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteRoleRequest = exports.DeleteRoleRequest || (exports.DeleteRoleRequest = {}));
var DeleteRolePermissionsBoundaryRequest;
(function (DeleteRolePermissionsBoundaryRequest) {
    DeleteRolePermissionsBoundaryRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteRolePermissionsBoundaryRequest = exports.DeleteRolePermissionsBoundaryRequest || (exports.DeleteRolePermissionsBoundaryRequest = {}));
var DeleteRolePolicyRequest;
(function (DeleteRolePolicyRequest) {
    DeleteRolePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteRolePolicyRequest = exports.DeleteRolePolicyRequest || (exports.DeleteRolePolicyRequest = {}));
var DeleteSAMLProviderRequest;
(function (DeleteSAMLProviderRequest) {
    DeleteSAMLProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSAMLProviderRequest = exports.DeleteSAMLProviderRequest || (exports.DeleteSAMLProviderRequest = {}));
var DeleteServerCertificateRequest;
(function (DeleteServerCertificateRequest) {
    DeleteServerCertificateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteServerCertificateRequest = exports.DeleteServerCertificateRequest || (exports.DeleteServerCertificateRequest = {}));
var DeleteServiceLinkedRoleRequest;
(function (DeleteServiceLinkedRoleRequest) {
    DeleteServiceLinkedRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteServiceLinkedRoleRequest = exports.DeleteServiceLinkedRoleRequest || (exports.DeleteServiceLinkedRoleRequest = {}));
var DeleteServiceLinkedRoleResponse;
(function (DeleteServiceLinkedRoleResponse) {
    DeleteServiceLinkedRoleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteServiceLinkedRoleResponse = exports.DeleteServiceLinkedRoleResponse || (exports.DeleteServiceLinkedRoleResponse = {}));
var DeleteServiceSpecificCredentialRequest;
(function (DeleteServiceSpecificCredentialRequest) {
    DeleteServiceSpecificCredentialRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteServiceSpecificCredentialRequest = exports.DeleteServiceSpecificCredentialRequest || (exports.DeleteServiceSpecificCredentialRequest = {}));
var DeleteSigningCertificateRequest;
(function (DeleteSigningCertificateRequest) {
    DeleteSigningCertificateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSigningCertificateRequest = exports.DeleteSigningCertificateRequest || (exports.DeleteSigningCertificateRequest = {}));
var DeleteSSHPublicKeyRequest;
(function (DeleteSSHPublicKeyRequest) {
    DeleteSSHPublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSSHPublicKeyRequest = exports.DeleteSSHPublicKeyRequest || (exports.DeleteSSHPublicKeyRequest = {}));
var DeleteUserRequest;
(function (DeleteUserRequest) {
    DeleteUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteUserRequest = exports.DeleteUserRequest || (exports.DeleteUserRequest = {}));
var DeleteUserPermissionsBoundaryRequest;
(function (DeleteUserPermissionsBoundaryRequest) {
    DeleteUserPermissionsBoundaryRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteUserPermissionsBoundaryRequest = exports.DeleteUserPermissionsBoundaryRequest || (exports.DeleteUserPermissionsBoundaryRequest = {}));
var DeleteUserPolicyRequest;
(function (DeleteUserPolicyRequest) {
    DeleteUserPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteUserPolicyRequest = exports.DeleteUserPolicyRequest || (exports.DeleteUserPolicyRequest = {}));
var DeleteVirtualMFADeviceRequest;
(function (DeleteVirtualMFADeviceRequest) {
    DeleteVirtualMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteVirtualMFADeviceRequest = exports.DeleteVirtualMFADeviceRequest || (exports.DeleteVirtualMFADeviceRequest = {}));
var DetachGroupPolicyRequest;
(function (DetachGroupPolicyRequest) {
    DetachGroupPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachGroupPolicyRequest = exports.DetachGroupPolicyRequest || (exports.DetachGroupPolicyRequest = {}));
var DetachRolePolicyRequest;
(function (DetachRolePolicyRequest) {
    DetachRolePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachRolePolicyRequest = exports.DetachRolePolicyRequest || (exports.DetachRolePolicyRequest = {}));
var DetachUserPolicyRequest;
(function (DetachUserPolicyRequest) {
    DetachUserPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachUserPolicyRequest = exports.DetachUserPolicyRequest || (exports.DetachUserPolicyRequest = {}));
var EnableMFADeviceRequest;
(function (EnableMFADeviceRequest) {
    EnableMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EnableMFADeviceRequest = exports.EnableMFADeviceRequest || (exports.EnableMFADeviceRequest = {}));
var InvalidAuthenticationCodeException;
(function (InvalidAuthenticationCodeException) {
    InvalidAuthenticationCodeException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidAuthenticationCodeException = exports.InvalidAuthenticationCodeException || (exports.InvalidAuthenticationCodeException = {}));
var ReportStateType;
(function (ReportStateType) {
    ReportStateType["COMPLETE"] = "COMPLETE";
    ReportStateType["INPROGRESS"] = "INPROGRESS";
    ReportStateType["STARTED"] = "STARTED";
})(ReportStateType = exports.ReportStateType || (exports.ReportStateType = {}));
var GenerateCredentialReportResponse;
(function (GenerateCredentialReportResponse) {
    GenerateCredentialReportResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateCredentialReportResponse = exports.GenerateCredentialReportResponse || (exports.GenerateCredentialReportResponse = {}));
var GenerateOrganizationsAccessReportRequest;
(function (GenerateOrganizationsAccessReportRequest) {
    GenerateOrganizationsAccessReportRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateOrganizationsAccessReportRequest = exports.GenerateOrganizationsAccessReportRequest || (exports.GenerateOrganizationsAccessReportRequest = {}));
var GenerateOrganizationsAccessReportResponse;
(function (GenerateOrganizationsAccessReportResponse) {
    GenerateOrganizationsAccessReportResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateOrganizationsAccessReportResponse = exports.GenerateOrganizationsAccessReportResponse || (exports.GenerateOrganizationsAccessReportResponse = {}));
var ReportGenerationLimitExceededException;
(function (ReportGenerationLimitExceededException) {
    ReportGenerationLimitExceededException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReportGenerationLimitExceededException = exports.ReportGenerationLimitExceededException || (exports.ReportGenerationLimitExceededException = {}));
var GenerateServiceLastAccessedDetailsRequest;
(function (GenerateServiceLastAccessedDetailsRequest) {
    GenerateServiceLastAccessedDetailsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateServiceLastAccessedDetailsRequest = exports.GenerateServiceLastAccessedDetailsRequest || (exports.GenerateServiceLastAccessedDetailsRequest = {}));
var GenerateServiceLastAccessedDetailsResponse;
(function (GenerateServiceLastAccessedDetailsResponse) {
    GenerateServiceLastAccessedDetailsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateServiceLastAccessedDetailsResponse = exports.GenerateServiceLastAccessedDetailsResponse || (exports.GenerateServiceLastAccessedDetailsResponse = {}));
var GetAccessKeyLastUsedRequest;
(function (GetAccessKeyLastUsedRequest) {
    GetAccessKeyLastUsedRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccessKeyLastUsedRequest = exports.GetAccessKeyLastUsedRequest || (exports.GetAccessKeyLastUsedRequest = {}));
var GetAccessKeyLastUsedResponse;
(function (GetAccessKeyLastUsedResponse) {
    GetAccessKeyLastUsedResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccessKeyLastUsedResponse = exports.GetAccessKeyLastUsedResponse || (exports.GetAccessKeyLastUsedResponse = {}));
var EntityType;
(function (EntityType) {
    EntityType["AWSManagedPolicy"] = "AWSManagedPolicy";
    EntityType["Group"] = "Group";
    EntityType["LocalManagedPolicy"] = "LocalManagedPolicy";
    EntityType["Role"] = "Role";
    EntityType["User"] = "User";
})(EntityType = exports.EntityType || (exports.EntityType = {}));
var GetAccountAuthorizationDetailsRequest;
(function (GetAccountAuthorizationDetailsRequest) {
    GetAccountAuthorizationDetailsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountAuthorizationDetailsRequest = exports.GetAccountAuthorizationDetailsRequest || (exports.GetAccountAuthorizationDetailsRequest = {}));
var PolicyDetail;
(function (PolicyDetail) {
    PolicyDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyDetail = exports.PolicyDetail || (exports.PolicyDetail = {}));
var GroupDetail;
(function (GroupDetail) {
    GroupDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GroupDetail = exports.GroupDetail || (exports.GroupDetail = {}));
var ManagedPolicyDetail;
(function (ManagedPolicyDetail) {
    ManagedPolicyDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ManagedPolicyDetail = exports.ManagedPolicyDetail || (exports.ManagedPolicyDetail = {}));
var RoleDetail;
(function (RoleDetail) {
    RoleDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RoleDetail = exports.RoleDetail || (exports.RoleDetail = {}));
var UserDetail;
(function (UserDetail) {
    UserDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UserDetail = exports.UserDetail || (exports.UserDetail = {}));
var GetAccountAuthorizationDetailsResponse;
(function (GetAccountAuthorizationDetailsResponse) {
    GetAccountAuthorizationDetailsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountAuthorizationDetailsResponse = exports.GetAccountAuthorizationDetailsResponse || (exports.GetAccountAuthorizationDetailsResponse = {}));
var PasswordPolicy;
(function (PasswordPolicy) {
    PasswordPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PasswordPolicy = exports.PasswordPolicy || (exports.PasswordPolicy = {}));
var GetAccountPasswordPolicyResponse;
(function (GetAccountPasswordPolicyResponse) {
    GetAccountPasswordPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountPasswordPolicyResponse = exports.GetAccountPasswordPolicyResponse || (exports.GetAccountPasswordPolicyResponse = {}));
var GetAccountSummaryResponse;
(function (GetAccountSummaryResponse) {
    GetAccountSummaryResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountSummaryResponse = exports.GetAccountSummaryResponse || (exports.GetAccountSummaryResponse = {}));
var GetContextKeysForCustomPolicyRequest;
(function (GetContextKeysForCustomPolicyRequest) {
    GetContextKeysForCustomPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetContextKeysForCustomPolicyRequest = exports.GetContextKeysForCustomPolicyRequest || (exports.GetContextKeysForCustomPolicyRequest = {}));
var GetContextKeysForPolicyResponse;
(function (GetContextKeysForPolicyResponse) {
    GetContextKeysForPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetContextKeysForPolicyResponse = exports.GetContextKeysForPolicyResponse || (exports.GetContextKeysForPolicyResponse = {}));
var GetContextKeysForPrincipalPolicyRequest;
(function (GetContextKeysForPrincipalPolicyRequest) {
    GetContextKeysForPrincipalPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetContextKeysForPrincipalPolicyRequest = exports.GetContextKeysForPrincipalPolicyRequest || (exports.GetContextKeysForPrincipalPolicyRequest = {}));
var CredentialReportExpiredException;
(function (CredentialReportExpiredException) {
    CredentialReportExpiredException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CredentialReportExpiredException = exports.CredentialReportExpiredException || (exports.CredentialReportExpiredException = {}));
var CredentialReportNotPresentException;
(function (CredentialReportNotPresentException) {
    CredentialReportNotPresentException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CredentialReportNotPresentException = exports.CredentialReportNotPresentException || (exports.CredentialReportNotPresentException = {}));
var CredentialReportNotReadyException;
(function (CredentialReportNotReadyException) {
    CredentialReportNotReadyException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CredentialReportNotReadyException = exports.CredentialReportNotReadyException || (exports.CredentialReportNotReadyException = {}));
var GetCredentialReportResponse;
(function (GetCredentialReportResponse) {
    GetCredentialReportResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCredentialReportResponse = exports.GetCredentialReportResponse || (exports.GetCredentialReportResponse = {}));
var GetGroupRequest;
(function (GetGroupRequest) {
    GetGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetGroupRequest = exports.GetGroupRequest || (exports.GetGroupRequest = {}));
var GetGroupResponse;
(function (GetGroupResponse) {
    GetGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetGroupResponse = exports.GetGroupResponse || (exports.GetGroupResponse = {}));
var GetGroupPolicyRequest;
(function (GetGroupPolicyRequest) {
    GetGroupPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetGroupPolicyRequest = exports.GetGroupPolicyRequest || (exports.GetGroupPolicyRequest = {}));
var GetGroupPolicyResponse;
(function (GetGroupPolicyResponse) {
    GetGroupPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetGroupPolicyResponse = exports.GetGroupPolicyResponse || (exports.GetGroupPolicyResponse = {}));
var GetInstanceProfileRequest;
(function (GetInstanceProfileRequest) {
    GetInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetInstanceProfileRequest = exports.GetInstanceProfileRequest || (exports.GetInstanceProfileRequest = {}));
var GetInstanceProfileResponse;
(function (GetInstanceProfileResponse) {
    GetInstanceProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetInstanceProfileResponse = exports.GetInstanceProfileResponse || (exports.GetInstanceProfileResponse = {}));
var GetLoginProfileRequest;
(function (GetLoginProfileRequest) {
    GetLoginProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLoginProfileRequest = exports.GetLoginProfileRequest || (exports.GetLoginProfileRequest = {}));
var GetLoginProfileResponse;
(function (GetLoginProfileResponse) {
    GetLoginProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLoginProfileResponse = exports.GetLoginProfileResponse || (exports.GetLoginProfileResponse = {}));
var GetOpenIDConnectProviderRequest;
(function (GetOpenIDConnectProviderRequest) {
    GetOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOpenIDConnectProviderRequest = exports.GetOpenIDConnectProviderRequest || (exports.GetOpenIDConnectProviderRequest = {}));
var GetOpenIDConnectProviderResponse;
(function (GetOpenIDConnectProviderResponse) {
    GetOpenIDConnectProviderResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOpenIDConnectProviderResponse = exports.GetOpenIDConnectProviderResponse || (exports.GetOpenIDConnectProviderResponse = {}));
var SortKeyType;
(function (SortKeyType) {
    SortKeyType["LAST_AUTHENTICATED_TIME_ASCENDING"] = "LAST_AUTHENTICATED_TIME_ASCENDING";
    SortKeyType["LAST_AUTHENTICATED_TIME_DESCENDING"] = "LAST_AUTHENTICATED_TIME_DESCENDING";
    SortKeyType["SERVICE_NAMESPACE_ASCENDING"] = "SERVICE_NAMESPACE_ASCENDING";
    SortKeyType["SERVICE_NAMESPACE_DESCENDING"] = "SERVICE_NAMESPACE_DESCENDING";
})(SortKeyType = exports.SortKeyType || (exports.SortKeyType = {}));
var GetOrganizationsAccessReportRequest;
(function (GetOrganizationsAccessReportRequest) {
    GetOrganizationsAccessReportRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOrganizationsAccessReportRequest = exports.GetOrganizationsAccessReportRequest || (exports.GetOrganizationsAccessReportRequest = {}));
var ErrorDetails;
(function (ErrorDetails) {
    ErrorDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ErrorDetails = exports.ErrorDetails || (exports.ErrorDetails = {}));
var GetOrganizationsAccessReportResponse;
(function (GetOrganizationsAccessReportResponse) {
    GetOrganizationsAccessReportResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOrganizationsAccessReportResponse = exports.GetOrganizationsAccessReportResponse || (exports.GetOrganizationsAccessReportResponse = {}));
var GetPolicyRequest;
(function (GetPolicyRequest) {
    GetPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyRequest = exports.GetPolicyRequest || (exports.GetPolicyRequest = {}));
var GetPolicyResponse;
(function (GetPolicyResponse) {
    GetPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyResponse = exports.GetPolicyResponse || (exports.GetPolicyResponse = {}));
var GetPolicyVersionRequest;
(function (GetPolicyVersionRequest) {
    GetPolicyVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyVersionRequest = exports.GetPolicyVersionRequest || (exports.GetPolicyVersionRequest = {}));
var GetPolicyVersionResponse;
(function (GetPolicyVersionResponse) {
    GetPolicyVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyVersionResponse = exports.GetPolicyVersionResponse || (exports.GetPolicyVersionResponse = {}));
var GetRoleRequest;
(function (GetRoleRequest) {
    GetRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRoleRequest = exports.GetRoleRequest || (exports.GetRoleRequest = {}));
var GetRoleResponse;
(function (GetRoleResponse) {
    GetRoleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRoleResponse = exports.GetRoleResponse || (exports.GetRoleResponse = {}));
var GetRolePolicyRequest;
(function (GetRolePolicyRequest) {
    GetRolePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRolePolicyRequest = exports.GetRolePolicyRequest || (exports.GetRolePolicyRequest = {}));
var GetRolePolicyResponse;
(function (GetRolePolicyResponse) {
    GetRolePolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRolePolicyResponse = exports.GetRolePolicyResponse || (exports.GetRolePolicyResponse = {}));
var GetSAMLProviderRequest;
(function (GetSAMLProviderRequest) {
    GetSAMLProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetSAMLProviderRequest = exports.GetSAMLProviderRequest || (exports.GetSAMLProviderRequest = {}));
var GetSAMLProviderResponse;
(function (GetSAMLProviderResponse) {
    GetSAMLProviderResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetSAMLProviderResponse = exports.GetSAMLProviderResponse || (exports.GetSAMLProviderResponse = {}));
var GetServerCertificateRequest;
(function (GetServerCertificateRequest) {
    GetServerCertificateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServerCertificateRequest = exports.GetServerCertificateRequest || (exports.GetServerCertificateRequest = {}));
var ServerCertificateMetadata;
(function (ServerCertificateMetadata) {
    ServerCertificateMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServerCertificateMetadata = exports.ServerCertificateMetadata || (exports.ServerCertificateMetadata = {}));
var ServerCertificate;
(function (ServerCertificate) {
    ServerCertificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServerCertificate = exports.ServerCertificate || (exports.ServerCertificate = {}));
var GetServerCertificateResponse;
(function (GetServerCertificateResponse) {
    GetServerCertificateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServerCertificateResponse = exports.GetServerCertificateResponse || (exports.GetServerCertificateResponse = {}));
var GetServiceLastAccessedDetailsRequest;
(function (GetServiceLastAccessedDetailsRequest) {
    GetServiceLastAccessedDetailsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServiceLastAccessedDetailsRequest = exports.GetServiceLastAccessedDetailsRequest || (exports.GetServiceLastAccessedDetailsRequest = {}));
var TrackedActionLastAccessed;
(function (TrackedActionLastAccessed) {
    TrackedActionLastAccessed.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TrackedActionLastAccessed = exports.TrackedActionLastAccessed || (exports.TrackedActionLastAccessed = {}));
var ServiceLastAccessed;
(function (ServiceLastAccessed) {
    ServiceLastAccessed.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServiceLastAccessed = exports.ServiceLastAccessed || (exports.ServiceLastAccessed = {}));
var GetServiceLastAccessedDetailsResponse;
(function (GetServiceLastAccessedDetailsResponse) {
    GetServiceLastAccessedDetailsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServiceLastAccessedDetailsResponse = exports.GetServiceLastAccessedDetailsResponse || (exports.GetServiceLastAccessedDetailsResponse = {}));
var GetServiceLastAccessedDetailsWithEntitiesRequest;
(function (GetServiceLastAccessedDetailsWithEntitiesRequest) {
    GetServiceLastAccessedDetailsWithEntitiesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServiceLastAccessedDetailsWithEntitiesRequest = exports.GetServiceLastAccessedDetailsWithEntitiesRequest || (exports.GetServiceLastAccessedDetailsWithEntitiesRequest = {}));
var PolicyOwnerEntityType;
(function (PolicyOwnerEntityType) {
    PolicyOwnerEntityType["GROUP"] = "GROUP";
    PolicyOwnerEntityType["ROLE"] = "ROLE";
    PolicyOwnerEntityType["USER"] = "USER";
})(PolicyOwnerEntityType = exports.PolicyOwnerEntityType || (exports.PolicyOwnerEntityType = {}));
var EntityInfo;
(function (EntityInfo) {
    EntityInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EntityInfo = exports.EntityInfo || (exports.EntityInfo = {}));
var EntityDetails;
(function (EntityDetails) {
    EntityDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EntityDetails = exports.EntityDetails || (exports.EntityDetails = {}));
var GetServiceLastAccessedDetailsWithEntitiesResponse;
(function (GetServiceLastAccessedDetailsWithEntitiesResponse) {
    GetServiceLastAccessedDetailsWithEntitiesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServiceLastAccessedDetailsWithEntitiesResponse = exports.GetServiceLastAccessedDetailsWithEntitiesResponse || (exports.GetServiceLastAccessedDetailsWithEntitiesResponse = {}));
var GetServiceLinkedRoleDeletionStatusRequest;
(function (GetServiceLinkedRoleDeletionStatusRequest) {
    GetServiceLinkedRoleDeletionStatusRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServiceLinkedRoleDeletionStatusRequest = exports.GetServiceLinkedRoleDeletionStatusRequest || (exports.GetServiceLinkedRoleDeletionStatusRequest = {}));
var RoleUsageType;
(function (RoleUsageType) {
    RoleUsageType.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RoleUsageType = exports.RoleUsageType || (exports.RoleUsageType = {}));
var DeletionTaskFailureReasonType;
(function (DeletionTaskFailureReasonType) {
    DeletionTaskFailureReasonType.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeletionTaskFailureReasonType = exports.DeletionTaskFailureReasonType || (exports.DeletionTaskFailureReasonType = {}));
var DeletionTaskStatusType;
(function (DeletionTaskStatusType) {
    DeletionTaskStatusType["FAILED"] = "FAILED";
    DeletionTaskStatusType["IN_PROGRESS"] = "IN_PROGRESS";
    DeletionTaskStatusType["NOT_STARTED"] = "NOT_STARTED";
    DeletionTaskStatusType["SUCCEEDED"] = "SUCCEEDED";
})(DeletionTaskStatusType = exports.DeletionTaskStatusType || (exports.DeletionTaskStatusType = {}));
var GetServiceLinkedRoleDeletionStatusResponse;
(function (GetServiceLinkedRoleDeletionStatusResponse) {
    GetServiceLinkedRoleDeletionStatusResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetServiceLinkedRoleDeletionStatusResponse = exports.GetServiceLinkedRoleDeletionStatusResponse || (exports.GetServiceLinkedRoleDeletionStatusResponse = {}));
var GetSSHPublicKeyRequest;
(function (GetSSHPublicKeyRequest) {
    GetSSHPublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetSSHPublicKeyRequest = exports.GetSSHPublicKeyRequest || (exports.GetSSHPublicKeyRequest = {}));
var SSHPublicKey;
(function (SSHPublicKey) {
    SSHPublicKey.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SSHPublicKey = exports.SSHPublicKey || (exports.SSHPublicKey = {}));
var GetSSHPublicKeyResponse;
(function (GetSSHPublicKeyResponse) {
    GetSSHPublicKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetSSHPublicKeyResponse = exports.GetSSHPublicKeyResponse || (exports.GetSSHPublicKeyResponse = {}));
var UnrecognizedPublicKeyEncodingException;
(function (UnrecognizedPublicKeyEncodingException) {
    UnrecognizedPublicKeyEncodingException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UnrecognizedPublicKeyEncodingException = exports.UnrecognizedPublicKeyEncodingException || (exports.UnrecognizedPublicKeyEncodingException = {}));
var GetUserRequest;
(function (GetUserRequest) {
    GetUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetUserRequest = exports.GetUserRequest || (exports.GetUserRequest = {}));
var GetUserResponse;
(function (GetUserResponse) {
    GetUserResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetUserResponse = exports.GetUserResponse || (exports.GetUserResponse = {}));
var GetUserPolicyRequest;
(function (GetUserPolicyRequest) {
    GetUserPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetUserPolicyRequest = exports.GetUserPolicyRequest || (exports.GetUserPolicyRequest = {}));
var GetUserPolicyResponse;
(function (GetUserPolicyResponse) {
    GetUserPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetUserPolicyResponse = exports.GetUserPolicyResponse || (exports.GetUserPolicyResponse = {}));
var ListAccessKeysRequest;
(function (ListAccessKeysRequest) {
    ListAccessKeysRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAccessKeysRequest = exports.ListAccessKeysRequest || (exports.ListAccessKeysRequest = {}));
var ListAccessKeysResponse;
(function (ListAccessKeysResponse) {
    ListAccessKeysResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAccessKeysResponse = exports.ListAccessKeysResponse || (exports.ListAccessKeysResponse = {}));
var ListAccountAliasesRequest;
(function (ListAccountAliasesRequest) {
    ListAccountAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAccountAliasesRequest = exports.ListAccountAliasesRequest || (exports.ListAccountAliasesRequest = {}));
var ListAccountAliasesResponse;
(function (ListAccountAliasesResponse) {
    ListAccountAliasesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAccountAliasesResponse = exports.ListAccountAliasesResponse || (exports.ListAccountAliasesResponse = {}));
var ListAttachedGroupPoliciesRequest;
(function (ListAttachedGroupPoliciesRequest) {
    ListAttachedGroupPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedGroupPoliciesRequest = exports.ListAttachedGroupPoliciesRequest || (exports.ListAttachedGroupPoliciesRequest = {}));
var ListAttachedGroupPoliciesResponse;
(function (ListAttachedGroupPoliciesResponse) {
    ListAttachedGroupPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedGroupPoliciesResponse = exports.ListAttachedGroupPoliciesResponse || (exports.ListAttachedGroupPoliciesResponse = {}));
var ListAttachedRolePoliciesRequest;
(function (ListAttachedRolePoliciesRequest) {
    ListAttachedRolePoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedRolePoliciesRequest = exports.ListAttachedRolePoliciesRequest || (exports.ListAttachedRolePoliciesRequest = {}));
var ListAttachedRolePoliciesResponse;
(function (ListAttachedRolePoliciesResponse) {
    ListAttachedRolePoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedRolePoliciesResponse = exports.ListAttachedRolePoliciesResponse || (exports.ListAttachedRolePoliciesResponse = {}));
var ListAttachedUserPoliciesRequest;
(function (ListAttachedUserPoliciesRequest) {
    ListAttachedUserPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedUserPoliciesRequest = exports.ListAttachedUserPoliciesRequest || (exports.ListAttachedUserPoliciesRequest = {}));
var ListAttachedUserPoliciesResponse;
(function (ListAttachedUserPoliciesResponse) {
    ListAttachedUserPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedUserPoliciesResponse = exports.ListAttachedUserPoliciesResponse || (exports.ListAttachedUserPoliciesResponse = {}));
var ListEntitiesForPolicyRequest;
(function (ListEntitiesForPolicyRequest) {
    ListEntitiesForPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListEntitiesForPolicyRequest = exports.ListEntitiesForPolicyRequest || (exports.ListEntitiesForPolicyRequest = {}));
var PolicyGroup;
(function (PolicyGroup) {
    PolicyGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyGroup = exports.PolicyGroup || (exports.PolicyGroup = {}));
var PolicyRole;
(function (PolicyRole) {
    PolicyRole.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyRole = exports.PolicyRole || (exports.PolicyRole = {}));
var PolicyUser;
(function (PolicyUser) {
    PolicyUser.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyUser = exports.PolicyUser || (exports.PolicyUser = {}));
var ListEntitiesForPolicyResponse;
(function (ListEntitiesForPolicyResponse) {
    ListEntitiesForPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListEntitiesForPolicyResponse = exports.ListEntitiesForPolicyResponse || (exports.ListEntitiesForPolicyResponse = {}));
var ListGroupPoliciesRequest;
(function (ListGroupPoliciesRequest) {
    ListGroupPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupPoliciesRequest = exports.ListGroupPoliciesRequest || (exports.ListGroupPoliciesRequest = {}));
var ListGroupPoliciesResponse;
(function (ListGroupPoliciesResponse) {
    ListGroupPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupPoliciesResponse = exports.ListGroupPoliciesResponse || (exports.ListGroupPoliciesResponse = {}));
var ListGroupsRequest;
(function (ListGroupsRequest) {
    ListGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupsRequest = exports.ListGroupsRequest || (exports.ListGroupsRequest = {}));
var ListGroupsResponse;
(function (ListGroupsResponse) {
    ListGroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupsResponse = exports.ListGroupsResponse || (exports.ListGroupsResponse = {}));
var ListGroupsForUserRequest;
(function (ListGroupsForUserRequest) {
    ListGroupsForUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupsForUserRequest = exports.ListGroupsForUserRequest || (exports.ListGroupsForUserRequest = {}));
var ListGroupsForUserResponse;
(function (ListGroupsForUserResponse) {
    ListGroupsForUserResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupsForUserResponse = exports.ListGroupsForUserResponse || (exports.ListGroupsForUserResponse = {}));
var ListInstanceProfilesRequest;
(function (ListInstanceProfilesRequest) {
    ListInstanceProfilesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInstanceProfilesRequest = exports.ListInstanceProfilesRequest || (exports.ListInstanceProfilesRequest = {}));
var ListInstanceProfilesResponse;
(function (ListInstanceProfilesResponse) {
    ListInstanceProfilesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInstanceProfilesResponse = exports.ListInstanceProfilesResponse || (exports.ListInstanceProfilesResponse = {}));
var ListInstanceProfilesForRoleRequest;
(function (ListInstanceProfilesForRoleRequest) {
    ListInstanceProfilesForRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInstanceProfilesForRoleRequest = exports.ListInstanceProfilesForRoleRequest || (exports.ListInstanceProfilesForRoleRequest = {}));
var ListInstanceProfilesForRoleResponse;
(function (ListInstanceProfilesForRoleResponse) {
    ListInstanceProfilesForRoleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInstanceProfilesForRoleResponse = exports.ListInstanceProfilesForRoleResponse || (exports.ListInstanceProfilesForRoleResponse = {}));
var ListInstanceProfileTagsRequest;
(function (ListInstanceProfileTagsRequest) {
    ListInstanceProfileTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInstanceProfileTagsRequest = exports.ListInstanceProfileTagsRequest || (exports.ListInstanceProfileTagsRequest = {}));
var ListInstanceProfileTagsResponse;
(function (ListInstanceProfileTagsResponse) {
    ListInstanceProfileTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInstanceProfileTagsResponse = exports.ListInstanceProfileTagsResponse || (exports.ListInstanceProfileTagsResponse = {}));
var ListMFADevicesRequest;
(function (ListMFADevicesRequest) {
    ListMFADevicesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListMFADevicesRequest = exports.ListMFADevicesRequest || (exports.ListMFADevicesRequest = {}));
var MFADevice;
(function (MFADevice) {
    MFADevice.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MFADevice = exports.MFADevice || (exports.MFADevice = {}));
var ListMFADevicesResponse;
(function (ListMFADevicesResponse) {
    ListMFADevicesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListMFADevicesResponse = exports.ListMFADevicesResponse || (exports.ListMFADevicesResponse = {}));
var ListMFADeviceTagsRequest;
(function (ListMFADeviceTagsRequest) {
    ListMFADeviceTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListMFADeviceTagsRequest = exports.ListMFADeviceTagsRequest || (exports.ListMFADeviceTagsRequest = {}));
var ListMFADeviceTagsResponse;
(function (ListMFADeviceTagsResponse) {
    ListMFADeviceTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListMFADeviceTagsResponse = exports.ListMFADeviceTagsResponse || (exports.ListMFADeviceTagsResponse = {}));
var ListOpenIDConnectProvidersRequest;
(function (ListOpenIDConnectProvidersRequest) {
    ListOpenIDConnectProvidersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOpenIDConnectProvidersRequest = exports.ListOpenIDConnectProvidersRequest || (exports.ListOpenIDConnectProvidersRequest = {}));
var OpenIDConnectProviderListEntry;
(function (OpenIDConnectProviderListEntry) {
    OpenIDConnectProviderListEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OpenIDConnectProviderListEntry = exports.OpenIDConnectProviderListEntry || (exports.OpenIDConnectProviderListEntry = {}));
var ListOpenIDConnectProvidersResponse;
(function (ListOpenIDConnectProvidersResponse) {
    ListOpenIDConnectProvidersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOpenIDConnectProvidersResponse = exports.ListOpenIDConnectProvidersResponse || (exports.ListOpenIDConnectProvidersResponse = {}));
var ListOpenIDConnectProviderTagsRequest;
(function (ListOpenIDConnectProviderTagsRequest) {
    ListOpenIDConnectProviderTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOpenIDConnectProviderTagsRequest = exports.ListOpenIDConnectProviderTagsRequest || (exports.ListOpenIDConnectProviderTagsRequest = {}));
var ListOpenIDConnectProviderTagsResponse;
(function (ListOpenIDConnectProviderTagsResponse) {
    ListOpenIDConnectProviderTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOpenIDConnectProviderTagsResponse = exports.ListOpenIDConnectProviderTagsResponse || (exports.ListOpenIDConnectProviderTagsResponse = {}));
var PolicyScopeType;
(function (PolicyScopeType) {
    PolicyScopeType["AWS"] = "AWS";
    PolicyScopeType["All"] = "All";
    PolicyScopeType["Local"] = "Local";
})(PolicyScopeType = exports.PolicyScopeType || (exports.PolicyScopeType = {}));
var ListPoliciesRequest;
(function (ListPoliciesRequest) {
    ListPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesRequest = exports.ListPoliciesRequest || (exports.ListPoliciesRequest = {}));
var ListPoliciesResponse;
(function (ListPoliciesResponse) {
    ListPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesResponse = exports.ListPoliciesResponse || (exports.ListPoliciesResponse = {}));
var ListPoliciesGrantingServiceAccessRequest;
(function (ListPoliciesGrantingServiceAccessRequest) {
    ListPoliciesGrantingServiceAccessRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesGrantingServiceAccessRequest = exports.ListPoliciesGrantingServiceAccessRequest || (exports.ListPoliciesGrantingServiceAccessRequest = {}));
var PolicyType;
(function (PolicyType) {
    PolicyType["INLINE"] = "INLINE";
    PolicyType["MANAGED"] = "MANAGED";
})(PolicyType = exports.PolicyType || (exports.PolicyType = {}));
var PolicyGrantingServiceAccess;
(function (PolicyGrantingServiceAccess) {
    PolicyGrantingServiceAccess.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyGrantingServiceAccess = exports.PolicyGrantingServiceAccess || (exports.PolicyGrantingServiceAccess = {}));
var ListPoliciesGrantingServiceAccessEntry;
(function (ListPoliciesGrantingServiceAccessEntry) {
    ListPoliciesGrantingServiceAccessEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesGrantingServiceAccessEntry = exports.ListPoliciesGrantingServiceAccessEntry || (exports.ListPoliciesGrantingServiceAccessEntry = {}));
var ListPoliciesGrantingServiceAccessResponse;
(function (ListPoliciesGrantingServiceAccessResponse) {
    ListPoliciesGrantingServiceAccessResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesGrantingServiceAccessResponse = exports.ListPoliciesGrantingServiceAccessResponse || (exports.ListPoliciesGrantingServiceAccessResponse = {}));
var ListPolicyTagsRequest;
(function (ListPolicyTagsRequest) {
    ListPolicyTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyTagsRequest = exports.ListPolicyTagsRequest || (exports.ListPolicyTagsRequest = {}));
var ListPolicyTagsResponse;
(function (ListPolicyTagsResponse) {
    ListPolicyTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyTagsResponse = exports.ListPolicyTagsResponse || (exports.ListPolicyTagsResponse = {}));
var ListPolicyVersionsRequest;
(function (ListPolicyVersionsRequest) {
    ListPolicyVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyVersionsRequest = exports.ListPolicyVersionsRequest || (exports.ListPolicyVersionsRequest = {}));
var ListPolicyVersionsResponse;
(function (ListPolicyVersionsResponse) {
    ListPolicyVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyVersionsResponse = exports.ListPolicyVersionsResponse || (exports.ListPolicyVersionsResponse = {}));
var ListRolePoliciesRequest;
(function (ListRolePoliciesRequest) {
    ListRolePoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRolePoliciesRequest = exports.ListRolePoliciesRequest || (exports.ListRolePoliciesRequest = {}));
var ListRolePoliciesResponse;
(function (ListRolePoliciesResponse) {
    ListRolePoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRolePoliciesResponse = exports.ListRolePoliciesResponse || (exports.ListRolePoliciesResponse = {}));
var ListRolesRequest;
(function (ListRolesRequest) {
    ListRolesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRolesRequest = exports.ListRolesRequest || (exports.ListRolesRequest = {}));
var ListRolesResponse;
(function (ListRolesResponse) {
    ListRolesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRolesResponse = exports.ListRolesResponse || (exports.ListRolesResponse = {}));
var ListRoleTagsRequest;
(function (ListRoleTagsRequest) {
    ListRoleTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRoleTagsRequest = exports.ListRoleTagsRequest || (exports.ListRoleTagsRequest = {}));
var ListRoleTagsResponse;
(function (ListRoleTagsResponse) {
    ListRoleTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRoleTagsResponse = exports.ListRoleTagsResponse || (exports.ListRoleTagsResponse = {}));
var ListSAMLProvidersRequest;
(function (ListSAMLProvidersRequest) {
    ListSAMLProvidersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSAMLProvidersRequest = exports.ListSAMLProvidersRequest || (exports.ListSAMLProvidersRequest = {}));
var SAMLProviderListEntry;
(function (SAMLProviderListEntry) {
    SAMLProviderListEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SAMLProviderListEntry = exports.SAMLProviderListEntry || (exports.SAMLProviderListEntry = {}));
var ListSAMLProvidersResponse;
(function (ListSAMLProvidersResponse) {
    ListSAMLProvidersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSAMLProvidersResponse = exports.ListSAMLProvidersResponse || (exports.ListSAMLProvidersResponse = {}));
var ListSAMLProviderTagsRequest;
(function (ListSAMLProviderTagsRequest) {
    ListSAMLProviderTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSAMLProviderTagsRequest = exports.ListSAMLProviderTagsRequest || (exports.ListSAMLProviderTagsRequest = {}));
var ListSAMLProviderTagsResponse;
(function (ListSAMLProviderTagsResponse) {
    ListSAMLProviderTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSAMLProviderTagsResponse = exports.ListSAMLProviderTagsResponse || (exports.ListSAMLProviderTagsResponse = {}));
var ListServerCertificatesRequest;
(function (ListServerCertificatesRequest) {
    ListServerCertificatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListServerCertificatesRequest = exports.ListServerCertificatesRequest || (exports.ListServerCertificatesRequest = {}));
var ListServerCertificatesResponse;
(function (ListServerCertificatesResponse) {
    ListServerCertificatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListServerCertificatesResponse = exports.ListServerCertificatesResponse || (exports.ListServerCertificatesResponse = {}));
var ListServerCertificateTagsRequest;
(function (ListServerCertificateTagsRequest) {
    ListServerCertificateTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListServerCertificateTagsRequest = exports.ListServerCertificateTagsRequest || (exports.ListServerCertificateTagsRequest = {}));
var ListServerCertificateTagsResponse;
(function (ListServerCertificateTagsResponse) {
    ListServerCertificateTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListServerCertificateTagsResponse = exports.ListServerCertificateTagsResponse || (exports.ListServerCertificateTagsResponse = {}));
var ListServiceSpecificCredentialsRequest;
(function (ListServiceSpecificCredentialsRequest) {
    ListServiceSpecificCredentialsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListServiceSpecificCredentialsRequest = exports.ListServiceSpecificCredentialsRequest || (exports.ListServiceSpecificCredentialsRequest = {}));
var ServiceSpecificCredentialMetadata;
(function (ServiceSpecificCredentialMetadata) {
    ServiceSpecificCredentialMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServiceSpecificCredentialMetadata = exports.ServiceSpecificCredentialMetadata || (exports.ServiceSpecificCredentialMetadata = {}));
var ListServiceSpecificCredentialsResponse;
(function (ListServiceSpecificCredentialsResponse) {
    ListServiceSpecificCredentialsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListServiceSpecificCredentialsResponse = exports.ListServiceSpecificCredentialsResponse || (exports.ListServiceSpecificCredentialsResponse = {}));
var ListSigningCertificatesRequest;
(function (ListSigningCertificatesRequest) {
    ListSigningCertificatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSigningCertificatesRequest = exports.ListSigningCertificatesRequest || (exports.ListSigningCertificatesRequest = {}));
var SigningCertificate;
(function (SigningCertificate) {
    SigningCertificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SigningCertificate = exports.SigningCertificate || (exports.SigningCertificate = {}));
var ListSigningCertificatesResponse;
(function (ListSigningCertificatesResponse) {
    ListSigningCertificatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSigningCertificatesResponse = exports.ListSigningCertificatesResponse || (exports.ListSigningCertificatesResponse = {}));
var ListSSHPublicKeysRequest;
(function (ListSSHPublicKeysRequest) {
    ListSSHPublicKeysRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSSHPublicKeysRequest = exports.ListSSHPublicKeysRequest || (exports.ListSSHPublicKeysRequest = {}));
var SSHPublicKeyMetadata;
(function (SSHPublicKeyMetadata) {
    SSHPublicKeyMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SSHPublicKeyMetadata = exports.SSHPublicKeyMetadata || (exports.SSHPublicKeyMetadata = {}));
var ListSSHPublicKeysResponse;
(function (ListSSHPublicKeysResponse) {
    ListSSHPublicKeysResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSSHPublicKeysResponse = exports.ListSSHPublicKeysResponse || (exports.ListSSHPublicKeysResponse = {}));
var ListUserPoliciesRequest;
(function (ListUserPoliciesRequest) {
    ListUserPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUserPoliciesRequest = exports.ListUserPoliciesRequest || (exports.ListUserPoliciesRequest = {}));
var ListUserPoliciesResponse;
(function (ListUserPoliciesResponse) {
    ListUserPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUserPoliciesResponse = exports.ListUserPoliciesResponse || (exports.ListUserPoliciesResponse = {}));
var ListUsersRequest;
(function (ListUsersRequest) {
    ListUsersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUsersRequest = exports.ListUsersRequest || (exports.ListUsersRequest = {}));
var ListUsersResponse;
(function (ListUsersResponse) {
    ListUsersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUsersResponse = exports.ListUsersResponse || (exports.ListUsersResponse = {}));
var ListUserTagsRequest;
(function (ListUserTagsRequest) {
    ListUserTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUserTagsRequest = exports.ListUserTagsRequest || (exports.ListUserTagsRequest = {}));
var ListUserTagsResponse;
(function (ListUserTagsResponse) {
    ListUserTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUserTagsResponse = exports.ListUserTagsResponse || (exports.ListUserTagsResponse = {}));
var ListVirtualMFADevicesRequest;
(function (ListVirtualMFADevicesRequest) {
    ListVirtualMFADevicesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListVirtualMFADevicesRequest = exports.ListVirtualMFADevicesRequest || (exports.ListVirtualMFADevicesRequest = {}));
var ListVirtualMFADevicesResponse;
(function (ListVirtualMFADevicesResponse) {
    ListVirtualMFADevicesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.VirtualMFADevices && {
            VirtualMFADevices: obj.VirtualMFADevices.map((item) => VirtualMFADevice.filterSensitiveLog(item)),
        }),
    });
})(ListVirtualMFADevicesResponse = exports.ListVirtualMFADevicesResponse || (exports.ListVirtualMFADevicesResponse = {}));
var PutGroupPolicyRequest;
(function (PutGroupPolicyRequest) {
    PutGroupPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutGroupPolicyRequest = exports.PutGroupPolicyRequest || (exports.PutGroupPolicyRequest = {}));
var PutRolePermissionsBoundaryRequest;
(function (PutRolePermissionsBoundaryRequest) {
    PutRolePermissionsBoundaryRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutRolePermissionsBoundaryRequest = exports.PutRolePermissionsBoundaryRequest || (exports.PutRolePermissionsBoundaryRequest = {}));
var PutRolePolicyRequest;
(function (PutRolePolicyRequest) {
    PutRolePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutRolePolicyRequest = exports.PutRolePolicyRequest || (exports.PutRolePolicyRequest = {}));
var PutUserPermissionsBoundaryRequest;
(function (PutUserPermissionsBoundaryRequest) {
    PutUserPermissionsBoundaryRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutUserPermissionsBoundaryRequest = exports.PutUserPermissionsBoundaryRequest || (exports.PutUserPermissionsBoundaryRequest = {}));
var PutUserPolicyRequest;
(function (PutUserPolicyRequest) {
    PutUserPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutUserPolicyRequest = exports.PutUserPolicyRequest || (exports.PutUserPolicyRequest = {}));
var RemoveClientIDFromOpenIDConnectProviderRequest;
(function (RemoveClientIDFromOpenIDConnectProviderRequest) {
    RemoveClientIDFromOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RemoveClientIDFromOpenIDConnectProviderRequest = exports.RemoveClientIDFromOpenIDConnectProviderRequest || (exports.RemoveClientIDFromOpenIDConnectProviderRequest = {}));
var RemoveRoleFromInstanceProfileRequest;
(function (RemoveRoleFromInstanceProfileRequest) {
    RemoveRoleFromInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RemoveRoleFromInstanceProfileRequest = exports.RemoveRoleFromInstanceProfileRequest || (exports.RemoveRoleFromInstanceProfileRequest = {}));
var RemoveUserFromGroupRequest;
(function (RemoveUserFromGroupRequest) {
    RemoveUserFromGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RemoveUserFromGroupRequest = exports.RemoveUserFromGroupRequest || (exports.RemoveUserFromGroupRequest = {}));
var ResetServiceSpecificCredentialRequest;
(function (ResetServiceSpecificCredentialRequest) {
    ResetServiceSpecificCredentialRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResetServiceSpecificCredentialRequest = exports.ResetServiceSpecificCredentialRequest || (exports.ResetServiceSpecificCredentialRequest = {}));
var ResetServiceSpecificCredentialResponse;
(function (ResetServiceSpecificCredentialResponse) {
    ResetServiceSpecificCredentialResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ServiceSpecificCredential && {
            ServiceSpecificCredential: ServiceSpecificCredential.filterSensitiveLog(obj.ServiceSpecificCredential),
        }),
    });
})(ResetServiceSpecificCredentialResponse = exports.ResetServiceSpecificCredentialResponse || (exports.ResetServiceSpecificCredentialResponse = {}));
var ResyncMFADeviceRequest;
(function (ResyncMFADeviceRequest) {
    ResyncMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResyncMFADeviceRequest = exports.ResyncMFADeviceRequest || (exports.ResyncMFADeviceRequest = {}));
var SetDefaultPolicyVersionRequest;
(function (SetDefaultPolicyVersionRequest) {
    SetDefaultPolicyVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetDefaultPolicyVersionRequest = exports.SetDefaultPolicyVersionRequest || (exports.SetDefaultPolicyVersionRequest = {}));
var GlobalEndpointTokenVersion;
(function (GlobalEndpointTokenVersion) {
    GlobalEndpointTokenVersion["v1Token"] = "v1Token";
    GlobalEndpointTokenVersion["v2Token"] = "v2Token";
})(GlobalEndpointTokenVersion = exports.GlobalEndpointTokenVersion || (exports.GlobalEndpointTokenVersion = {}));
var SetSecurityTokenServicePreferencesRequest;
(function (SetSecurityTokenServicePreferencesRequest) {
    SetSecurityTokenServicePreferencesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetSecurityTokenServicePreferencesRequest = exports.SetSecurityTokenServicePreferencesRequest || (exports.SetSecurityTokenServicePreferencesRequest = {}));
var PolicyEvaluationException;
(function (PolicyEvaluationException) {
    PolicyEvaluationException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyEvaluationException = exports.PolicyEvaluationException || (exports.PolicyEvaluationException = {}));
var ContextKeyTypeEnum;
(function (ContextKeyTypeEnum) {
    ContextKeyTypeEnum["BINARY"] = "binary";
    ContextKeyTypeEnum["BINARY_LIST"] = "binaryList";
    ContextKeyTypeEnum["BOOLEAN"] = "boolean";
    ContextKeyTypeEnum["BOOLEAN_LIST"] = "booleanList";
    ContextKeyTypeEnum["DATE"] = "date";
    ContextKeyTypeEnum["DATE_LIST"] = "dateList";
    ContextKeyTypeEnum["IP"] = "ip";
    ContextKeyTypeEnum["IP_LIST"] = "ipList";
    ContextKeyTypeEnum["NUMERIC"] = "numeric";
    ContextKeyTypeEnum["NUMERIC_LIST"] = "numericList";
    ContextKeyTypeEnum["STRING"] = "string";
    ContextKeyTypeEnum["STRING_LIST"] = "stringList";
})(ContextKeyTypeEnum = exports.ContextKeyTypeEnum || (exports.ContextKeyTypeEnum = {}));
var ContextEntry;
(function (ContextEntry) {
    ContextEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ContextEntry = exports.ContextEntry || (exports.ContextEntry = {}));
var SimulateCustomPolicyRequest;
(function (SimulateCustomPolicyRequest) {
    SimulateCustomPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SimulateCustomPolicyRequest = exports.SimulateCustomPolicyRequest || (exports.SimulateCustomPolicyRequest = {}));
var PolicyEvaluationDecisionType;
(function (PolicyEvaluationDecisionType) {
    PolicyEvaluationDecisionType["ALLOWED"] = "allowed";
    PolicyEvaluationDecisionType["EXPLICIT_DENY"] = "explicitDeny";
    PolicyEvaluationDecisionType["IMPLICIT_DENY"] = "implicitDeny";
})(PolicyEvaluationDecisionType = exports.PolicyEvaluationDecisionType || (exports.PolicyEvaluationDecisionType = {}));
var Position;
(function (Position) {
    Position.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Position = exports.Position || (exports.Position = {}));
var PolicySourceType;
(function (PolicySourceType) {
    PolicySourceType["AWS_MANAGED"] = "aws-managed";
    PolicySourceType["GROUP"] = "group";
    PolicySourceType["NONE"] = "none";
    PolicySourceType["RESOURCE"] = "resource";
    PolicySourceType["ROLE"] = "role";
    PolicySourceType["USER"] = "user";
    PolicySourceType["USER_MANAGED"] = "user-managed";
})(PolicySourceType = exports.PolicySourceType || (exports.PolicySourceType = {}));
var Statement;
(function (Statement) {
    Statement.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Statement = exports.Statement || (exports.Statement = {}));
var OrganizationsDecisionDetail;
(function (OrganizationsDecisionDetail) {
    OrganizationsDecisionDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OrganizationsDecisionDetail = exports.OrganizationsDecisionDetail || (exports.OrganizationsDecisionDetail = {}));
var PermissionsBoundaryDecisionDetail;
(function (PermissionsBoundaryDecisionDetail) {
    PermissionsBoundaryDecisionDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PermissionsBoundaryDecisionDetail = exports.PermissionsBoundaryDecisionDetail || (exports.PermissionsBoundaryDecisionDetail = {}));
var ResourceSpecificResult;
(function (ResourceSpecificResult) {
    ResourceSpecificResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceSpecificResult = exports.ResourceSpecificResult || (exports.ResourceSpecificResult = {}));
var EvaluationResult;
(function (EvaluationResult) {
    EvaluationResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EvaluationResult = exports.EvaluationResult || (exports.EvaluationResult = {}));
var SimulatePolicyResponse;
(function (SimulatePolicyResponse) {
    SimulatePolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SimulatePolicyResponse = exports.SimulatePolicyResponse || (exports.SimulatePolicyResponse = {}));
var SimulatePrincipalPolicyRequest;
(function (SimulatePrincipalPolicyRequest) {
    SimulatePrincipalPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SimulatePrincipalPolicyRequest = exports.SimulatePrincipalPolicyRequest || (exports.SimulatePrincipalPolicyRequest = {}));
var TagInstanceProfileRequest;
(function (TagInstanceProfileRequest) {
    TagInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagInstanceProfileRequest = exports.TagInstanceProfileRequest || (exports.TagInstanceProfileRequest = {}));
var TagMFADeviceRequest;
(function (TagMFADeviceRequest) {
    TagMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagMFADeviceRequest = exports.TagMFADeviceRequest || (exports.TagMFADeviceRequest = {}));
var TagOpenIDConnectProviderRequest;
(function (TagOpenIDConnectProviderRequest) {
    TagOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagOpenIDConnectProviderRequest = exports.TagOpenIDConnectProviderRequest || (exports.TagOpenIDConnectProviderRequest = {}));
var TagPolicyRequest;
(function (TagPolicyRequest) {
    TagPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagPolicyRequest = exports.TagPolicyRequest || (exports.TagPolicyRequest = {}));
var TagRoleRequest;
(function (TagRoleRequest) {
    TagRoleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagRoleRequest = exports.TagRoleRequest || (exports.TagRoleRequest = {}));
var TagSAMLProviderRequest;
(function (TagSAMLProviderRequest) {
    TagSAMLProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagSAMLProviderRequest = exports.TagSAMLProviderRequest || (exports.TagSAMLProviderRequest = {}));
var TagServerCertificateRequest;
(function (TagServerCertificateRequest) {
    TagServerCertificateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagServerCertificateRequest = exports.TagServerCertificateRequest || (exports.TagServerCertificateRequest = {}));
var TagUserRequest;
(function (TagUserRequest) {
    TagUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagUserRequest = exports.TagUserRequest || (exports.TagUserRequest = {}));
var UntagInstanceProfileRequest;
(function (UntagInstanceProfileRequest) {
    UntagInstanceProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagInstanceProfileRequest = exports.UntagInstanceProfileRequest || (exports.UntagInstanceProfileRequest = {}));
var UntagMFADeviceRequest;
(function (UntagMFADeviceRequest) {
    UntagMFADeviceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagMFADeviceRequest = exports.UntagMFADeviceRequest || (exports.UntagMFADeviceRequest = {}));
var UntagOpenIDConnectProviderRequest;
(function (UntagOpenIDConnectProviderRequest) {
    UntagOpenIDConnectProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagOpenIDConnectProviderRequest = exports.UntagOpenIDConnectProviderRequest || (exports.UntagOpenIDConnectProviderRequest = {}));
